<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Class notes – BIOL 7117: Speciation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6b3bef205c96c4f74d1e1b8d3d254fec.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIOL 7117: Speciation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./class-notes.html" aria-current="page"> 
<span class="menu-text">Class notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./readings.html"> 
<span class="menu-text">Reading notes</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#overview-species-concepts" id="toc-overview-species-concepts" class="nav-link active" data-scroll-target="#overview-species-concepts"><span class="header-section-number">1</span> Overview &amp; species concepts</a>
  <ul class="collapse">
  <li><a href="#course-overview" id="toc-course-overview" class="nav-link" data-scroll-target="#course-overview"><span class="header-section-number">1.1</span> Course overview</a></li>
  <li><a href="#concepts-of-species" id="toc-concepts-of-species" class="nav-link" data-scroll-target="#concepts-of-species"><span class="header-section-number">1.2</span> Concepts of species</a>
  <ul class="collapse">
  <li><a href="#biological-species-concept" id="toc-biological-species-concept" class="nav-link" data-scroll-target="#biological-species-concept"><span class="header-section-number">1.2.1</span> Biological species concept</a></li>
  <li><a href="#phylogenetic-species-concepts" id="toc-phylogenetic-species-concepts" class="nav-link" data-scroll-target="#phylogenetic-species-concepts"><span class="header-section-number">1.2.2</span> Phylogenetic species concepts</a></li>
  <li><a href="#van-valen-1976-ecological-species-concept" id="toc-van-valen-1976-ecological-species-concept" class="nav-link" data-scroll-target="#van-valen-1976-ecological-species-concept"><span class="header-section-number">1.2.3</span> van Valen 1976: Ecological species concept</a></li>
  <li><a href="#de-queiroz-1998-general-lineage-concept" id="toc-de-queiroz-1998-general-lineage-concept" class="nav-link" data-scroll-target="#de-queiroz-1998-general-lineage-concept"><span class="header-section-number">1.2.4</span> De Queiroz 1998: General lineage concept</a></li>
  <li><a href="#alternativedissenting-species-concepts" id="toc-alternativedissenting-species-concepts" class="nav-link" data-scroll-target="#alternativedissenting-species-concepts"><span class="header-section-number">1.2.5</span> Alternative/dissenting species concepts</a></li>
  <li><a href="#operational-definition" id="toc-operational-definition" class="nav-link" data-scroll-target="#operational-definition"><span class="header-section-number">1.2.6</span> Operational definition</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#geography-of-speciation" id="toc-geography-of-speciation" class="nav-link" data-scroll-target="#geography-of-speciation"><span class="header-section-number">2</span> Geography of speciation</a>
  <ul class="collapse">
  <li><a href="#allopatric-speciation" id="toc-allopatric-speciation" class="nav-link" data-scroll-target="#allopatric-speciation"><span class="header-section-number">2.1</span> Allopatric speciation</a></li>
  <li><a href="#sympatric-speciation" id="toc-sympatric-speciation" class="nav-link" data-scroll-target="#sympatric-speciation"><span class="header-section-number">2.2</span> Sympatric speciation</a>
  <ul class="collapse">
  <li><a href="#problems" id="toc-problems" class="nav-link" data-scroll-target="#problems"><span class="header-section-number">2.2.1</span> Problems</a></li>
  <li><a href="#solutions" id="toc-solutions" class="nav-link" data-scroll-target="#solutions"><span class="header-section-number">2.2.2</span> Solutions</a></li>
  <li><a href="#outstanding-problem" id="toc-outstanding-problem" class="nav-link" data-scroll-target="#outstanding-problem"><span class="header-section-number">2.2.3</span> Outstanding problem</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#discussion-geography-of-speciation" id="toc-discussion-geography-of-speciation" class="nav-link" data-scroll-target="#discussion-geography-of-speciation">Discussion: Geography of speciation</a></li>
  <li><a href="#prezygotic-isolation" id="toc-prezygotic-isolation" class="nav-link" data-scroll-target="#prezygotic-isolation"><span class="header-section-number">3</span> Prezygotic isolation</a>
  <ul class="collapse">
  <li><a href="#evidence-for-ecological-prezygotic-isolation" id="toc-evidence-for-ecological-prezygotic-isolation" class="nav-link" data-scroll-target="#evidence-for-ecological-prezygotic-isolation"><span class="header-section-number">3.1</span> Evidence for ecological prezygotic isolation</a></li>
  <li><a href="#evidence-for-sexual-prezygotic-isolation" id="toc-evidence-for-sexual-prezygotic-isolation" class="nav-link" data-scroll-target="#evidence-for-sexual-prezygotic-isolation"><span class="header-section-number">3.2</span> Evidence for sexual prezygotic isolation</a></li>
  </ul></li>
  <li><a href="#discussion-prezygotic-isolation" id="toc-discussion-prezygotic-isolation" class="nav-link" data-scroll-target="#discussion-prezygotic-isolation">Discussion: Prezygotic isolation</a></li>
  <li><a href="#postzygotic-isolation" id="toc-postzygotic-isolation" class="nav-link" data-scroll-target="#postzygotic-isolation"><span class="header-section-number">4</span> Postzygotic isolation</a>
  <ul class="collapse">
  <li><a href="#extrinsic-postzygotic-isolation" id="toc-extrinsic-postzygotic-isolation" class="nav-link" data-scroll-target="#extrinsic-postzygotic-isolation"><span class="header-section-number">4.1</span> Extrinsic postzygotic isolation</a></li>
  <li><a href="#intrinsic-postzygotic-isolation" id="toc-intrinsic-postzygotic-isolation" class="nav-link" data-scroll-target="#intrinsic-postzygotic-isolation"><span class="header-section-number">4.2</span> Intrinsic postzygotic isolation</a>
  <ul class="collapse">
  <li><a href="#haldanes-rule" id="toc-haldanes-rule" class="nav-link" data-scroll-target="#haldanes-rule"><span class="header-section-number">4.2.1</span> Haldane’s Rule</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#discussion-postzygotic-isolation" id="toc-discussion-postzygotic-isolation" class="nav-link" data-scroll-target="#discussion-postzygotic-isolation">Discussion: Postzygotic isolation</a></li>
  <li><a href="#reinforcement" id="toc-reinforcement" class="nav-link" data-scroll-target="#reinforcement"><span class="header-section-number">5</span> Reinforcement</a>
  <ul class="collapse">
  <li><a href="#experimental-tests-for-reinforcement" id="toc-experimental-tests-for-reinforcement" class="nav-link" data-scroll-target="#experimental-tests-for-reinforcement"><span class="header-section-number">5.1</span> Experimental tests for reinforcement</a></li>
  <li><a href="#comparative-studies-of-reinforcement" id="toc-comparative-studies-of-reinforcement" class="nav-link" data-scroll-target="#comparative-studies-of-reinforcement"><span class="header-section-number">5.2</span> Comparative studies of reinforcement</a></li>
  <li><a href="#objections-to-reinforcement" id="toc-objections-to-reinforcement" class="nav-link" data-scroll-target="#objections-to-reinforcement"><span class="header-section-number">5.3</span> Objections to reinforcement</a></li>
  <li><a href="#solutions-to-objections" id="toc-solutions-to-objections" class="nav-link" data-scroll-target="#solutions-to-objections"><span class="header-section-number">5.4</span> Solutions to objections</a></li>
  <li><a href="#bottom-line" id="toc-bottom-line" class="nav-link" data-scroll-target="#bottom-line"><span class="header-section-number">5.5</span> Bottom line</a></li>
  </ul></li>
  <li><a href="#wait-what-background-primer" id="toc-wait-what-background-primer" class="nav-link" data-scroll-target="#wait-what-background-primer"><span class="header-section-number">6</span> Wait, what? Background primer</a>
  <ul class="collapse">
  <li><a href="#historical-demography-of-speciation" id="toc-historical-demography-of-speciation" class="nav-link" data-scroll-target="#historical-demography-of-speciation"><span class="header-section-number">6.1</span> Historical demography of speciation</a></li>
  <li><a href="#measures-of-genetic-diversity" id="toc-measures-of-genetic-diversity" class="nav-link" data-scroll-target="#measures-of-genetic-diversity"><span class="header-section-number">6.2</span> Measures of genetic diversity</a></li>
  <li><a href="#isolation-migration-model-for-historical-demography" id="toc-isolation-migration-model-for-historical-demography" class="nav-link" data-scroll-target="#isolation-migration-model-for-historical-demography"><span class="header-section-number">6.3</span> Isolation-migration model for historical demography</a></li>
  <li><a href="#other-models" id="toc-other-models" class="nav-link" data-scroll-target="#other-models"><span class="header-section-number">6.4</span> Other models</a></li>
  <li><a href="#tests-for-selection" id="toc-tests-for-selection" class="nav-link" data-scroll-target="#tests-for-selection"><span class="header-section-number">6.5</span> Tests for selection</a></li>
  </ul></li>
  <li><a href="#genetic-basis-of-speciation-i-how-many-genes" id="toc-genetic-basis-of-speciation-i-how-many-genes" class="nav-link" data-scroll-target="#genetic-basis-of-speciation-i-how-many-genes"><span class="header-section-number">7</span> Genetic basis of speciation I: How many genes?</a>
  <ul class="collapse">
  <li><a href="#darwinian-gradualism-many-genes-of-small-effect" id="toc-darwinian-gradualism-many-genes-of-small-effect" class="nav-link" data-scroll-target="#darwinian-gradualism-many-genes-of-small-effect"><span class="header-section-number">7.1</span> Darwinian gradualism | many genes of small effect</a></li>
  <li><a href="#mendelian-saltationism-few-genes-of-large-effect" id="toc-mendelian-saltationism-few-genes-of-large-effect" class="nav-link" data-scroll-target="#mendelian-saltationism-few-genes-of-large-effect"><span class="header-section-number">7.2</span> Mendelian saltationism | few genes of large effect</a></li>
  <li><a href="#modern-synthesis" id="toc-modern-synthesis" class="nav-link" data-scroll-target="#modern-synthesis"><span class="header-section-number">7.3</span> Modern Synthesis</a></li>
  <li><a href="#which-genes" id="toc-which-genes" class="nav-link" data-scroll-target="#which-genes"><span class="header-section-number">7.4</span> Which genes?</a>
  <ul class="collapse">
  <li><a href="#forward-genetics" id="toc-forward-genetics" class="nav-link" data-scroll-target="#forward-genetics"><span class="header-section-number">7.4.1</span> Forward genetics</a></li>
  </ul></li>
  <li><a href="#candidate-genes" id="toc-candidate-genes" class="nav-link" data-scroll-target="#candidate-genes"><span class="header-section-number">7.5</span> Candidate genes</a></li>
  <li><a href="#reverse-genetics" id="toc-reverse-genetics" class="nav-link" data-scroll-target="#reverse-genetics"><span class="header-section-number">7.6</span> Reverse genetics</a></li>
  </ul></li>
  <li><a href="#discussion-genetic-basis-of-speciation-i" id="toc-discussion-genetic-basis-of-speciation-i" class="nav-link" data-scroll-target="#discussion-genetic-basis-of-speciation-i">Discussion: Genetic basis of speciation I</a></li>
  <li><a href="#genetic-basis-of-speciation-ii-which-genes" id="toc-genetic-basis-of-speciation-ii-which-genes" class="nav-link" data-scroll-target="#genetic-basis-of-speciation-ii-which-genes"><span class="header-section-number">8</span> Genetic basis of speciation II: Which genes?</a>
  <ul class="collapse">
  <li><a href="#structural-vs-regulatory-sequences" id="toc-structural-vs-regulatory-sequences" class="nav-link" data-scroll-target="#structural-vs-regulatory-sequences"><span class="header-section-number">8.1</span> Structural vs regulatory sequences</a></li>
  <li><a href="#cis-vs-trans-mechanism-of-regulation" id="toc-cis-vs-trans-mechanism-of-regulation" class="nav-link" data-scroll-target="#cis-vs-trans-mechanism-of-regulation"><span class="header-section-number">8.2</span> <em>cis</em> vs <em>trans</em> | mechanism of regulation</a></li>
  <li><a href="#cis-regulatory-changes-structural-changes" id="toc-cis-regulatory-changes-structural-changes" class="nav-link" data-scroll-target="#cis-regulatory-changes-structural-changes"><span class="header-section-number">8.3</span> cis-regulatory changes &gt; structural changes</a></li>
  <li><a href="#in-defence-of-structural-changes" id="toc-in-defence-of-structural-changes" class="nav-link" data-scroll-target="#in-defence-of-structural-changes"><span class="header-section-number">8.4</span> In defence of structural changes</a></li>
  </ul></li>
  <li><a href="#discussion-genetic-basis-of-speciation-ii-gene-regulation" id="toc-discussion-genetic-basis-of-speciation-ii-gene-regulation" class="nav-link" data-scroll-target="#discussion-genetic-basis-of-speciation-ii-gene-regulation">Discussion: Genetic basis of speciation II (Gene regulation)</a></li>
  <li><a href="#genomic-architecture" id="toc-genomic-architecture" class="nav-link" data-scroll-target="#genomic-architecture"><span class="header-section-number">9</span> Genomic architecture</a>
  <ul class="collapse">
  <li><a href="#genomic-islands-of-speciation-differentiation" id="toc-genomic-islands-of-speciation-differentiation" class="nav-link" data-scroll-target="#genomic-islands-of-speciation-differentiation"><span class="header-section-number">9.1</span> Genomic islands of speciation (differentiation)</a></li>
  <li><a href="#disputed-islands" id="toc-disputed-islands" class="nav-link" data-scroll-target="#disputed-islands"><span class="header-section-number">9.2</span> Disputed islands</a></li>
  </ul></li>
  <li><a href="#discussion-genomic-architecture" id="toc-discussion-genomic-architecture" class="nav-link" data-scroll-target="#discussion-genomic-architecture">Discussion: Genomic architecture</a></li>
  <li><a href="#hybridisation-and-introgression" id="toc-hybridisation-and-introgression" class="nav-link" data-scroll-target="#hybridisation-and-introgression"><span class="header-section-number">10</span> Hybridisation and introgression</a>
  <ul class="collapse">
  <li><a href="#fusion" id="toc-fusion" class="nav-link" data-scroll-target="#fusion"><span class="header-section-number">10.1</span> Fusion</a></li>
  <li><a href="#hybrid-speciation" id="toc-hybrid-speciation" class="nav-link" data-scroll-target="#hybrid-speciation"><span class="header-section-number">10.2</span> Hybrid speciation</a></li>
  <li><a href="#balance" id="toc-balance" class="nav-link" data-scroll-target="#balance"><span class="header-section-number">10.3</span> Balance</a></li>
  <li><a href="#introgression" id="toc-introgression" class="nav-link" data-scroll-target="#introgression"><span class="header-section-number">10.4</span> Introgression</a></li>
  </ul></li>
  <li><a href="#discussion-introgression" id="toc-discussion-introgression" class="nav-link" data-scroll-target="#discussion-introgression">Discussion: Introgression</a></li>
  <li><a href="#old-variation-is-the-new-thing" id="toc-old-variation-is-the-new-thing" class="nav-link" data-scroll-target="#old-variation-is-the-new-thing"><span class="header-section-number">11</span> Old variation is the new thing</a>
  <ul class="collapse">
  <li><a href="#de-novo-variation" id="toc-de-novo-variation" class="nav-link" data-scroll-target="#de-novo-variation"><span class="header-section-number">11.1</span> <em>de novo</em> variation</a></li>
  <li><a href="#standing-variation" id="toc-standing-variation" class="nav-link" data-scroll-target="#standing-variation"><span class="header-section-number">11.2</span> Standing variation</a></li>
  <li><a href="#combinatorial-view-of-speciation" id="toc-combinatorial-view-of-speciation" class="nav-link" data-scroll-target="#combinatorial-view-of-speciation"><span class="header-section-number">11.3</span> Combinatorial view of speciation</a></li>
  </ul></li>
  <li><a href="#discussion-old-variation" id="toc-discussion-old-variation" class="nav-link" data-scroll-target="#discussion-old-variation">Discussion: Old variation</a></li>
  <li><a href="#microbes-microbiomes-and-speciation" id="toc-microbes-microbiomes-and-speciation" class="nav-link" data-scroll-target="#microbes-microbiomes-and-speciation"><span class="header-section-number">12</span> Microbes, microbiomes and speciation</a>
  <ul class="collapse">
  <li><a href="#species" id="toc-species" class="nav-link" data-scroll-target="#species"><span class="header-section-number">12.1</span> Species</a></li>
  <li><a href="#communities" id="toc-communities" class="nav-link" data-scroll-target="#communities"><span class="header-section-number">12.2</span> Communities</a></li>
  <li><a href="#holobiont" id="toc-holobiont" class="nav-link" data-scroll-target="#holobiont"><span class="header-section-number">12.3</span> Holobiont</a></li>
  </ul></li>
  <li><a href="#macroevolution" id="toc-macroevolution" class="nav-link" data-scroll-target="#macroevolution"><span class="header-section-number">13</span> Macroevolution</a>
  <ul class="collapse">
  <li><a href="#fossil-record" id="toc-fossil-record" class="nav-link" data-scroll-target="#fossil-record"><span class="header-section-number">13.1</span> Fossil record</a></li>
  <li><a href="#phylogenetic-trees" id="toc-phylogenetic-trees" class="nav-link" data-scroll-target="#phylogenetic-trees"><span class="header-section-number">13.2</span> Phylogenetic trees</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Class notes</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="overview-species-concepts" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Overview &amp; species concepts</h1>
<p><strong>2025-08-26</strong></p>
<section id="course-overview" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="course-overview"><span class="header-section-number">1.1</span> Course overview</h2>
<p>Follows Coyne &amp; Orr 2004(?) broadly. This was a highly impactful book, very timely, and heralded the dawn of the genomic era. The period of roughly 2012–20 saw incredible growth in our understanding of genomes and speciation. Species concepts are important, especially if your thesis includes the term “speciation”, but we won’t spend too much time in the class on these debates on definition.</p>
<p>Till exam 1: reading fundamentals of speciation questions. Post-exam-1: stick with basic questions but delve on what genomics has done to advance understanding. Word and sentence counts in assignments need to be followed strictly, aim is to get used to condensing important information.</p>
</section>
<section id="concepts-of-species" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="concepts-of-species"><span class="header-section-number">1.2</span> Concepts of species</h2>
<p>Species are not just for bookkeeping; different species are actually different, and they do different things. Cryptic species are an example of why (community) ecologists should care about work of population geneticists.</p>
<blockquote class="blockquote">
<p>If one could conclude as to the nature of the Creator from a study of creation it would appear that God has an inordinate fondness for stars and beetles.</p>
<p>— JBS Haldane, on creationism and the sheer number of beetle species</p>
</blockquote>
<p>Early ideas of species concepts can date back to, as with many things, Aristotle. His notion of a perfect “essence” asserted the existence of a perfect form for each species, variations from which are imperfections in its actualisation. The essence of a species was fixed, but involved descent as the crucial factor tying different individuals in a species together. This implied a fundamental difference between inter- and intraspecific variation.</p>
<section id="biological-species-concept" class="level3" data-number="1.2.1">
<h3 data-number="1.2.1" class="anchored" data-anchor-id="biological-species-concept"><span class="header-section-number">1.2.1</span> Biological species concept</h3>
<p>The <strong>Darwinian revolution</strong> insisted that variation is not just noise, but rather the core (nuisance vs crux). <em>Everything</em> shares common descent, <em>variation</em> is at the heart of evolution, and there is no true difference between inter- and intraspecific variation. His way of distinguishing variants was using the morphological gap criterion (few intermediates), although his concept can be applied to ecology, behaviour or genetics too.</p>
<p>Ernst Mayr, starting in the 1950s, adapted Darwin’s ideas to eventually come up with the biological species concept, where species are defined as interbreeding natural populations, reproductively isolated from others. Before him, Poulton (1904) had similar ideas, with the notion of “syngamy” (interbreeding) as the true meaning of species. (As did Lotsy and Dobzhansky.)</p>
<p>Some adherents of BSC include those who worked extensively on model organisms like <em>Drosophila</em> in the lab (Coyne, Orr, Noor).</p>
<p><strong>Impact of BSC</strong></p>
<ul>
<li>Species now defined by process important to their own maintenance (biological, as opposed to scientist sitting in a museum describing from specimens).</li>
<li>Allowed, and drew attention to, concepts of gene flow and isolating mechanisms.</li>
<li>Provided criterion for diagnosing cryptic taxa.</li>
</ul>
<p><strong>Problems of BSC</strong></p>
<p>Less universally applicable than Darwinian concept:</p>
<ul>
<li>Provides no useful insight regarding large-ranged species having several geographically isolated populations (difficult to check interbreeding in allopatry).</li>
<li>Hybridisation is rather common, even between non-sister taxa. This might not be just noise!</li>
<li>Cannot deal with asexual organisms that have divided into obviously recognisable populations/groups.</li>
<li>Circularity: diagnosis of species and cause of speciation are the same.</li>
</ul>
<p>Also, reproductive isolation not likely adaptive, and prezygotic compatibility is the true reality underlying species—isolating mechanisms a result rather than a cause of species separateness (Paterson 1985). <em>Recognition concept</em> vs isolation concept, common fertilisation system.</p>
</section>
<section id="phylogenetic-species-concepts" class="level3" data-number="1.2.2">
<h3 data-number="1.2.2" class="anchored" data-anchor-id="phylogenetic-species-concepts"><span class="header-section-number">1.2.2</span> Phylogenetic species concepts</h3>
<section id="hennig-1966-cladisticsmonophyly-based" class="level4" data-number="1.2.2.1">
<h4 data-number="1.2.2.1" class="anchored" data-anchor-id="hennig-1966-cladisticsmonophyly-based"><span class="header-section-number">1.2.2.1</span> Hennig 1966: Cladistics/monophyly-based</h4>
<p>Hennig brought about the discipline of <em>cladistics</em>, which eventually evolved into the modern PSC, where a clade means a monophyletic group. Therefore, this species concept is based on monophyly, i.e., history. Problems:</p>
<ul>
<li>Can lead to considering very small isolated groups (due to focus on monophyly)</li>
<li>Forced to leave behind paraphyletic taxa</li>
<li>Contingent on quality of trees! Early on, mostly mitochondrial DNA.</li>
<li>Species tree vs gene tree discordance.
<ul>
<li>Maddison 1997 showed how this discordance can arise (short speciation event times, wide branches?, related to incomplete lineage sorting).</li>
<li>Can’t rely blindly on numbers from gene trees.</li>
</ul></li>
</ul>
</section>
<section id="cracraft-1989-irreducible-clusterdiagnosably-distinct" class="level4" data-number="1.2.2.2">
<h4 data-number="1.2.2.2" class="anchored" data-anchor-id="cracraft-1989-irreducible-clusterdiagnosably-distinct"><span class="header-section-number">1.2.2.2</span> Cracraft 1989: Irreducible cluster/diagnosably distinct</h4>
<p>Focus on irreducible clusters of organisms, diagnosably distinct from others, and among which there is a parental pattern of ancestry and descent. Most proponents of this were from a museum background. Recently, this concept has started to make inroads into zoological taxonomy, raising subspecies previously demoted (under polytypic species concept) to species level.</p>
</section>
<section id="modern-phylogenetic-approaches" class="level4" data-number="1.2.2.3">
<h4 data-number="1.2.2.3" class="anchored" data-anchor-id="modern-phylogenetic-approaches"><span class="header-section-number">1.2.2.3</span> Modern phylogenetic approaches</h4>
<ul>
<li>Species delimitations using <strong>multi-locus coalescent</strong> (Yang &amp; Rannala 2014); using Bayesian posterior to compare single-species or two-species models based on genomic data
<ul>
<li>Problems: Same as Bayesian methods in general—importance of prior info on phylogenetic relationships.</li>
</ul></li>
<li><strong>Barcode of Life</strong> (Paul Hebert): driven by biodiversity crisis, don’t need to wait for taxonomic expertise, can instead just collect and insert in barcoding pipeline.
<ul>
<li>Problems: For many species, there is no clean gap in barcodes to distinguish intra- and interspecific variation. So, useful for identifying taxa from samples, but controversial for describing new species.</li>
<li>Also, single gene taxonomy, gene-vs-species tree, introgression of mtDNA</li>
</ul></li>
</ul>
</section>
</section>
<section id="van-valen-1976-ecological-species-concept" class="level3" data-number="1.2.3">
<h3 data-number="1.2.3" class="anchored" data-anchor-id="van-valen-1976-ecological-species-concept"><span class="header-section-number">1.2.3</span> van Valen 1976: Ecological species concept</h3>
<p>Places greater focus on stabilising selection than gene flow. This is an especially true/useful approach for bacteria: different bacteria can occupy very different niches and have no gene flow, but act/behave the same in terms of where they are found, physiology, etc.</p>
</section>
<section id="de-queiroz-1998-general-lineage-concept" class="level3" data-number="1.2.4">
<h3 data-number="1.2.4" class="anchored" data-anchor-id="de-queiroz-1998-general-lineage-concept"><span class="header-section-number">1.2.4</span> De Queiroz 1998: General lineage concept</h3>
<p>Focuses on evolutionary independence, and considers species as segments of population-level variation?</p>
</section>
<section id="alternativedissenting-species-concepts" class="level3" data-number="1.2.5">
<h3 data-number="1.2.5" class="anchored" data-anchor-id="alternativedissenting-species-concepts"><span class="header-section-number">1.2.5</span> Alternative/dissenting species concepts</h3>
<p><strong>Species are not real!</strong></p>
<p>Some perspectives/justifications for this view:</p>
<ul>
<li>Taxonomic perspective: subspecies may be elevated to species, but still grade into each other so don’t hold much meaning</li>
<li>Consider populations as evolutionary units, not species</li>
<li>Spatial and temporal forms that are not easy to tell apart as species vs not.</li>
<li>Genetic markers: gene flow too low to maintain cohesion</li>
<li>Phenetics</li>
</ul>
</section>
<section id="operational-definition" class="level3" data-number="1.2.6">
<h3 data-number="1.2.6" class="anchored" data-anchor-id="operational-definition"><span class="header-section-number">1.2.6</span> Operational definition</h3>
<p>Hellberg’s operational definition is based on <strong>genotypic cluster criterion (Mallet 1995)</strong>. The emphasis is on how combinations of traits cluster separately or together between groups. What’s critical is the presence of genotypic gaps between local populations, so this concept makes sense for sympatric populations but not so much for allopatric ones.</p>
<p>It can be pictured as a bimodal distribution of traits. This view relies on Bayesian stats, using “assignment tests” and so on.</p>
</section>
</section>
</section>
<section id="geography-of-speciation" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Geography of speciation</h1>
<p><strong>2025-08-28</strong>: Allopatric and sympatric speciation.</p>
<p>Allopatric speciation has always been popular, and has been the preferred model of speciation until recently, whereas sympatric speciation has kinda been the underdog.</p>
<section id="allopatric-speciation" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="allopatric-speciation"><span class="header-section-number">2.1</span> Allopatric speciation</h2>
<p><strong>Jordan’s Law (1907):</strong> David Starr Jordan was the founder of Stanford University. His model system was Indopacific wrasses, and he had ideas about allopatric speciation well before Ernst Mayr. He believed: given any species in any region, nearest related species not likely to be found in the same region but in a neighbouring district separated from the first by a barrier of some sort. A demonstrative example was squirrels in the Grand Canyon.</p>
<p>Ernst Mayr was a big proponent of allopatry, and specifically peripatry. He emphasised the role of drift in populations. Model systems were birds of paradise and <em>Dicrurus</em> in SE Asia.</p>
<p>This has been the preferred model for multiple reasons:</p>
<ul>
<li>Easier: no gene flow to contend with</li>
<li>Lots of opportunities in nature for allopatry</li>
<li>Lots of evidence:
<ul>
<li>Young sister species tend to be allopatric</li>
<li>Concordance between geography and species borders</li>
<li>Extension of above: Geographic coincidence of multiple separations of sister species and/or hybrid zones (one barrier/feature showing up again separating multiple groups of species)</li>
<li>Increase in reproductive isolation in allopatry</li>
</ul></li>
</ul>
<p>However, even in this model, not incorporating explicit selection (i.e., relying only on drift/mutation) into calculations of species generation times results in very long times.</p>
</section>
<section id="sympatric-speciation" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="sympatric-speciation"><span class="header-section-number">2.2</span> Sympatric speciation</h2>
<section id="problems" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="problems"><span class="header-section-number">2.2.1</span> Problems</h3>
<p><strong>Genetic problem:</strong> Felsenstein (1981)—also inventor of bootstrapping, maximum likelihood estimation, etc.; has 15 papers with &gt;1000 citations—argued for the problem of recombination. In sympatry, need to avoid recombination by mating differentially.</p>
<p><strong>Ecological problem:</strong> Ecological patterns like coexistence, competitive exclusion (Gause’s Rule) occur because in sympatry need to live differently enough from others.</p>
</section>
<section id="solutions" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="solutions"><span class="header-section-number">2.2.2</span> Solutions</h3>
<p>Disruptive sexual selection; disruptive natural selection (mating is linked to habitat, and the fewer genes [of large effect] the better). These are now very much linked to the concept of <strong>magic traits</strong>, for which there has been much evidence.</p>
</section>
<section id="outstanding-problem" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="outstanding-problem"><span class="header-section-number">2.2.3</span> Outstanding problem</h3>
<p>Additional requirements for sympatric speciation are that they need to be sister species, and we need to be sure that past allopatry was unlikely. This has been more difficult to show, and many popular examples fail here.</p>
<p>Lake Victoria cichlids had crazy sympatric speciation possible due to a combination of factors like trophic divergence, sexual selection, and microallopatry. However, there is controversy around the idea that they are only 14,000 years old—some people don’t think the lake fully dried out.</p>
<p>In the case of <em>Rhagoletis</em> fruit flies, which have sympatric differentiation in host plant choice (between hawthorn and apple; GL Bush), it turned out that they were actually allopatric at some point and only the completion of the speciation process happened in sympatry (Feder et al 2003).</p>
<p>Similarly, in the case of stickleback ecomorphs in postglacial lakes, secondary contact drove character displacement and resulted in sister species, even though completion was in sympatry (Rundle &amp; Schluter 2004).</p>
<p>Thus, more and more evidence has piled up to transform the dichotomy from sympatric vs allopatric speciation, to <strong>primary sympatric divergence vs secondary contact introgression</strong>.</p>
</section>
</section>
</section>
<section id="discussion-geography-of-speciation" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Geography of speciation</h1>
<p><strong>2025-09-02</strong></p>
<ul>
<li>Doesn’t mean old school (primary sympatric divergence) is fully obsolete. In fact, palm paper is an example. Hellberg also believes it should be one model under consideration.</li>
<li>What % hybridisation is relevant gene flow? 5% is kinda high still (1 in 20).</li>
<li>Each data point in a genetic structure plot (k-means clustering) is the allele type expressed at each biallelic locus (genotype).</li>
<li><em>Differentiation</em> is different from <em>variation</em></li>
</ul>
</section>
<section id="prezygotic-isolation" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Prezygotic isolation</h1>
<p><strong>2025-09-04</strong></p>
<p>Reproductive isolation was a hot topic in evolution after WWII and well into the 1990s. Studies focused on two kinds of reproductive barriers (prezygotic and postzygotic), and eventually this dichotomy was paralleled in the scientists who studied them (field vs lab [e.g., <em>Drosophila</em> folks] respectively).</p>
<p>Prezygotic isolation can be considered in two broad, but not mutually exclusive, categories: ecological (temporal, pollinator, habitat) and sexual (behavioural, mechanical, gametic) isolation.</p>
<section id="evidence-for-ecological-prezygotic-isolation" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="evidence-for-ecological-prezygotic-isolation"><span class="header-section-number">3.1</span> Evidence for ecological prezygotic isolation</h2>
<p><strong>Parallel speciation</strong>: stickleback ecomorphs in post-glacial lakes; ecotypes that are geographically isolated still reproduce</p>
<p><strong>Magic traits</strong>: ecological gene under selection pleiotropically causes assortative mating (affects reproductive isolation). <em>Heliconius</em> butterflies of Amazonia; cichlids in E Africa lake (depth of lake affects colouration favoured); feather lice &amp; pigeon hosts (Villa et al 2019 PNAS; check <a href="https://www.pnas.org/doi/10.1073/pnas.1901247116#supplementary-materials">videos</a>).</p>
<p><strong>Strong habitat selection</strong>: walking sticks in California, immigrant inviability (abiotic-dispersed organisms)</p>
<p><strong>Isolation-by-distance/adaptation</strong>: geographic distance <span class="math inline">\(\longrightarrow\)</span> genetic distance OR ecological/adaptive divergence <span class="math inline">\(\longrightarrow\)</span> genetic distance</p>
</section>
<section id="evidence-for-sexual-prezygotic-isolation" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="evidence-for-sexual-prezygotic-isolation"><span class="header-section-number">3.2</span> Evidence for sexual prezygotic isolation</h2>
<p><strong>Runaway sexual selection</strong>: positive feedback loops</p>
<p><strong>Sexual antagonism</strong>: males and females may have different optima for same phenotype</p>
<p><strong>Mating isolation</strong>: genitalia differentiation in beetles, moths, etc.; lock-and-key also in pollinator/flower compatibility.</p>
<p><strong>Gametic isolation</strong> is a special case which is post-mating but pre-zygotic (reproductive tracts, bodily fluids). Conspecific sperm precedence (mating order); acrosome reaction for V envelope dissolution around eggs in marine gastropods.</p>
</section>
</section>
<section id="discussion-prezygotic-isolation" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Prezygotic isolation</h1>
<p><strong>2025-09-09</strong></p>
<ul>
<li>Sentences in Science are usually more assertive/confident than would be otherwise.</li>
<li>GWAS: genome-wide association studies.</li>
<li>Sex chromosomes limit damage when dangerous alleles show up (doesn’t work same in both sexes, sexually antagonistic)</li>
</ul>
</section>
<section id="postzygotic-isolation" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Postzygotic isolation</h1>
<p><strong>2025-09-11</strong></p>
<p>Barriers after the zygote is formed, which can act via reduced hybrid viability, reduced hybrid fertility, hybrid breakdown, etc.</p>
<p>This topic has received much less focus than prezygotic isolation, because of ascertainment bias: once prezygotic isolation is completed, then cannot see postzygotic isolation. Also, other things like reinforcement strengthen prezygotic isolation.</p>
<p>However, there are many reasons to consider postzygotic isolation. First, in many ways, <strong>ecological isolation</strong> is a form of postzygotic isolation. Moreover, even in cases where hybrid formation occurs, the persistence of distinct species means there is something <strong>decreasing the fitness of these hybrids</strong>. Also, <strong>intrinsic post- isolation may become stronger</strong> at <span class="math inline">\(\ge\)</span> F2 crosses, while extrinsic post- and pre- become weaker</p>
<p>Darwin had a conundrum about contradictory ways in which evolution worked: on the one hand natural selection operates, but on the other hand unfit hybrids are produced. His solution for this was that sterility arises as an incidental byprouct of “unknown differences”.</p>
<p>Much of the interest in postzygotic isolation is wrt intrinsic.</p>
<section id="extrinsic-postzygotic-isolation" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="extrinsic-postzygotic-isolation"><span class="header-section-number">4.1</span> Extrinsic postzygotic isolation</h2>
<p>External to the individual/organism, i.e., non-developmental. This includes things like habitat isolation, where say niche of hybrids falls somewhere between parental lineages but fitness of hybrids is lower than that of either parent. But in some cases, hybrids may have higher fitness in intermediate habitats, which would explain the increase in hybridisation in the first place.</p>
<p>In stickleback crossing studies, Rundle 2002 found that F1s were bad (relative to parents), but F1-parent backcrosses yielded some fitter individuals (proportional to how much parental heredity the cross had).</p>
<p>Other times, hybrids could have higher disease susceptibility, or intermediate traits that are less attractive (sexual selection).</p>
</section>
<section id="intrinsic-postzygotic-isolation" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="intrinsic-postzygotic-isolation"><span class="header-section-number">4.2</span> Intrinsic postzygotic isolation</h2>
<p>Internal to the individual/organism, so quite simply, developmental defects lead to hybrid sterility or inviability.</p>
<p>It can be <strong>meiosis issues</strong> related to ploidy levels, where hybrids like F1s simply cannot perform meiosis. Or <strong>chromosomal rearrangements/inversions</strong> like in Australian morabine grasshoppers (White 1978). However, many inversions surprisingly have no effect on heterozygote fitness, because recombination can still happen in other regions of genome.</p>
<p>Other mechanisms include <strong>endosymbiotic infection</strong> risk for hybrids, or <strong>cytoplasmic incompatibilities</strong> especially relating to parasites (e.g., intracellular parasites like <em>Wolbachia</em>: when uninfected female flies are crossed with infected males, many/all progeny die, but infection transmitted from mother so those that are born from infected females are infected). There can even be <strong>mitonuclear incompatibilities</strong> (i.e., between mitochrondrial and cytoplasmic genetic material), famously in copepods (Ron Burton), so much so that Geoffrey Hill even introduced a <em>mitonuclear compatibility species concept</em>.</p>
<p><strong>Allelic incompatibilities</strong> are a major way intrinsic postzygotic isolation can occur. The original definition of Dobzhansky-Muller Incompatibility focused on alleles at different loci, arguing that low hybrid fitness is a byproduct of genomes that are geographically isolated. The newer, more specific definition emphasises epistasis (effects of an allele at one locus depend on alleles present at other loci), arguing that <strong>negative epistatic</strong> interactions occur between alleles of different loci with different evolutionary histories. Some assumptions of this theory are that there occurs at least one pair of allele changes, that most incompatibilities are due to <strong>derived alleles</strong>, and that the no. if incompatible combinations snowballs over time.</p>
<section id="haldanes-rule" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="haldanes-rule"><span class="header-section-number">4.2.1</span> Haldane’s Rule</h3>
<p>Rule about <strong>sex chromosome incompatibilities</strong>; it is an exception, because there are very rare exceptions to the rule. Basically: in offspring of two different races, if one (and only one) sex is messed up (absent, rare, or sterile) then that is the heterozygous sex.</p>
<p><strong>NOT caused by:</strong> incompatibilities between X- and Y-linked genes; dosage compensation (compensate based on importance of stuff on X); chromosomal arrangement.</p>
<p>Possible causes:</p>
<ul>
<li><strong>Dominance</strong>: in XY species, hybrid males are affected by all X-linked genes involved in genic incompatibilities, whereas hybrid females are affected only by dominant X-linked genes</li>
<li><del><strong>Faster-males</strong>: in XY species, incompatibilities affecting XY hybrids accumulate more quickly</del> (but exceptions to this, so doesn’t satisfactorily explain)</li>
<li><strong>Faster-X</strong>: faster drift at population levels because more X than Y in population (thus this mechanism interacts with dominance)</li>
<li><strong>Meiotic drive</strong>: distorter alleles that distort Mendelian ratios to their own advantage; suppressors evolve to stop them. When two allopatric populations each independently evolve a driver and suppressor, suppressed drivers can become unmasked.</li>
</ul>
</section>
</section>
</section>
<section id="discussion-postzygotic-isolation" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Postzygotic isolation</h1>
<p><strong>2025-09-16</strong>: <em>(absent)</em></p>
</section>
<section id="reinforcement" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Reinforcement</h1>
<p><strong>2025-09-18</strong></p>
<p>Reinforcement has been explored in the context of driving enhanced prezygotic isolation in sympatry. It has traditionally defined as when the <strong>formation of maladaptive hybrids</strong> selects for <strong>enhanced prezyogtic isolation</strong>—where hybrids are maladaptive if their fitness is non-zero but lower than either parent’s. The late 1980s saw renewed interest in exploring such questions of reinforcement.</p>
<p>However, there was also a concurrent realisation that enhanced prezygotic isolation in sympatry could arise not just due to selection against hybrids, but also due to ecological or other factors. This brought focus to <strong>reproductive character displacement</strong>, where selection for enhanced prezygotic isolation occurs in order to reduce wasted reproductive effort between RI species. For instance, acoustic space divergence in frogs in TX/MX occurs between not just close congeners but also distant relatives.</p>
<p>Many early popular examples of reinforcement are actually of reproductive character displacement, because the hybrid has zero fitness (no hybrid formation in sympatry). Reviewed in Butlin 1987. Other cases, like the Collared and Pied flycatchers, do conform to the strictly reinforcement expectations, but we no know that there is more going on there.</p>
<section id="experimental-tests-for-reinforcement" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="experimental-tests-for-reinforcement"><span class="header-section-number">5.1</span> Experimental tests for reinforcement</h2>
<p>Reviewed in Rice &amp; Hostert 1993.</p>
<ul>
<li><strong>Kill-the-hybrid:</strong> let them mate but kill hybrids and see if displacement develops
<ul>
<li>But this is actually evidence of RCD than of reinforcement</li>
<li>Seemingly high success rate, but publication bias?</li>
</ul></li>
<li><strong>Disruptive selection on arbitrary trait:</strong> kill everything but high and low values of certain neutral trait (like bristles in <em>Drosophila</em>)
<ul>
<li>Bristles are not really neutral/arbitrary, have important function</li>
<li>On the other hand, these results were never replicated in further studies</li>
</ul></li>
</ul>
</section>
<section id="comparative-studies-of-reinforcement" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="comparative-studies-of-reinforcement"><span class="header-section-number">5.2</span> Comparative studies of reinforcement</h2>
<p>Can ask the following questions:</p>
<ul>
<li>Is prezyogtic isolation greater in sympatry?</li>
<li>(Difficult to test) Are they forming hybrids?</li>
<li>(Difficult to test) What is the genetic basis? e.g., does having many great effect genes at few nearby loci facilitate more reinforcement (magic traits)?</li>
</ul>
<p>Lots of work in this area, including on <em>Drosophila</em> from Dobzhansky &amp; Koller 1938 (western US) to Matute 2010 (Sao Tome &amp; Principe), and bindin protein in sea urchins. In the case of the <em>Ficedula</em> flycatchers, male plumage is driven by female choice (because females in sympatry are not different, only males are), and this is just females being choosier than males (fewer eggs than sperm). However, it is not fully clear whether the brown in Pied males evolved to enhance isolation from sympatric Collared, or whether it’s just female-mimicry in order to reduce intermale aggression. This is the <em>more going on</em> in this case.</p>
<p>The genetic basis of reinforcement has been reviewed in Garner et al 2018, and consists of genetic divergence both within species (allopatric vs sympatric contrasts) and between species (greater in sympatry, lower in allopatry). One example is flower colour in <em>Phlox</em> wildflowers, where colour is driven by two loci (intensity and hue), but selection happens on the intensity one.</p>
</section>
<section id="objections-to-reinforcement" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="objections-to-reinforcement"><span class="header-section-number">5.3</span> Objections to reinforcement</h2>
<ul>
<li>Need to reconcile selection and recombination (antagonism, similar to that in case of sympatric speciation), which gets worse with more interacting genes</li>
<li>Race against extinction: Have to be ecologically distinct to survive in hybrid zone, BUT not enough, need to stick around in hybrid zone (asking for a lot)</li>
<li>Swamping effect: genes for reinforcement must be favoured only in sympatry; genes from outside hybrid zone may swamp out (again, asking a lot)</li>
</ul>
</section>
<section id="solutions-to-objections" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="solutions-to-objections"><span class="header-section-number">5.4</span> Solutions to objections</h2>
<ul>
<li>Liou &amp; Price 1994: <strong>sexual selection</strong> can promote reinforcement in hybrid zone</li>
<li>Kelly &amp; Noor 1996: selection for greater female discrimination can promote reinforcement; “one-allele” solution (same allele favoured in both species); not just sexual selection but <strong>strength of selection</strong> (turn up volume/intensity)</li>
</ul>
</section>
<section id="bottom-line" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="bottom-line"><span class="header-section-number">5.5</span> Bottom line</h2>
<p>This is a theoretically possible mechanism, but in most cases it’s asking for a lot. There is a lot of evidence for greater divergence in sympatry, but most of these cases also have alternative explanations.</p>
<p>The best bet for such a mechanism would involve female choice resulting in heavier divergence in sympatry. This means that systems with greater sexual selection may be more likely to have reinforcement.</p>
</section>
</section>
<section id="wait-what-background-primer" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Wait, what? Background primer</h1>
<p><strong>2025-09-25</strong></p>
<section id="historical-demography-of-speciation" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="historical-demography-of-speciation"><span class="header-section-number">6.1</span> Historical demography of speciation</h2>
<p>Genetic studies of speciation often infer the species tree from gene trees. Divergence time for the genes is always older than for the species. Trees made with different genes may be different and have disagreements (is this a methods issue, or the actual population demographics, or are they important factors driving speciation itself?). Moreover, gene trees can be complicated by mixing after divergence. So all this variation needs to be accounted for.</p>
<p>Historical demography helps us infer about population size, changes in population size, divergence genetics, and isolation and hybridisation. In this demographic context, certain terms have specific meanings. Population size is considered an indication of genetic variation, with larger populations holding more genetic variation. Genetic diversity is commonly denoted as <span class="math inline">\(\theta\)</span>, but there are multiple metrics used.</p>
</section>
<section id="measures-of-genetic-diversity" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="measures-of-genetic-diversity"><span class="header-section-number">6.2</span> Measures of genetic diversity</h2>
<p><span class="math inline">\(\theta = 4 N_e u\)</span> is a simple metric representing expected genetic variation, where <span class="math inline">\(N_e\)</span> is the effective population size (size of an idealised population that would show the same degree of drift as observed), and <span class="math inline">\(u\)</span> is substitution rate (sometimes aka mutation rate).</p>
<p><strong>Watterson’s <span class="math inline">\(\theta_w\)</span></strong>: Ignores frequencies, and is strongly affected by rare alleles. Based on the number of segregating sites (polymorphisms) in a DNA sample.</p>
<p><strong>Tajima’s <span class="math inline">\(\theta_\pi\)</span></strong>: Based on frequency so rare alleles don’t matter. Observed nucleotide diversity: average number of differences between pairs of sequences in a DNA sample.</p>
<p><strong>Tajima’s <span class="math inline">\(D\)</span></strong> gives an overall measure. Premise is that under neutrality (observed diversity matching expected diversity), <span class="math inline">\(D = 0\)</span>. <span class="math inline">\(D \lt 0\)</span> means more rare alleles than expected (negative selection), suggesting bottleneck (multiple loci) or selective sweep (single locus). During selective sweep, when <span class="math inline">\(D\)</span> drops, linkage disequilibrium rises. <em>Biston betularia</em> is a classic example of selective sweep. <span class="math inline">\(D \gt 0\)</span> means few rare alleles, suggesting balancing selection (e.g., malaria resistance) or overdominant selection. If you follow a certain selective pressure over time, <span class="math inline">\(D\)</span> would go from negative to positive.</p>
<p><strong>Mismatch distribution</strong> (Rogers &amp; Harpending 1992): quantify mismatches between individuals in a population for each locus, and visualise like a histogram. A clustered peak shows sudden high mutation event; so we can estimate divergence time (e.g., bottleneck in exploding population). This requires examining non-recombining bits of DNA, and earlier mitochondrial DNA was used for this, but nowadays it’s the Y chromosome.</p>
<p><strong>Site/allele frequency spectrum</strong>: distribution of allele frequencies for all segregating sites in a sample of DNA sequences. Certain demographic scenarios like bottleneck would result in allele frequency distribution (using combined polymorphism data) close to 0 for 2nd most common allele (i.e., minor allele in biallelic). We can compare frequency with reference of outgroup (unfolded) to infer derived vs ancestral.</p>
<p><strong>Synteny</strong>: conservation of blocks of (gene) order between two sets of chromosomes being compared.</p>
</section>
<section id="isolation-migration-model-for-historical-demography" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="isolation-migration-model-for-historical-demography"><span class="header-section-number">6.3</span> Isolation-migration model for historical demography</h2>
<p>Developed by Jody Hey and others, as a way to address the problem of discordance between histories of gene and species trees. It is based on coalescent theory, where all the lineages present today coalesce going back to one common ancestor. Using only the present-day observed lineages helps because we can ignore others from past who didn’t make it (simplifies compute). Smaller (pop. size) and younger lineages would coalesce faster. We can also get gene history or gene geneaology along the same steps.</p>
<p>This can be used to test for <em>no isolation</em> between population, by looking for <span class="math inline">\(t = 0\)</span> (divergence time). Similarly, it can test for <em>no migration</em> (genetic exchange), if <span class="math inline">\(m_1 = m_2 = 0\)</span>. On the other hand, it can also estimate values for these parameters for conditions of isolation and migration.</p>
</section>
<section id="other-models" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="other-models"><span class="header-section-number">6.4</span> Other models</h2>
<p>The IM model is very simplistic, mainly because it assumes just one population size outside of divergence, and migration rate is constant. One alternative is <strong>DaDi</strong> (Diffusion Approximations for Demographic Inference), which allows more than one value for each parameter. However, this is a different framework, because it works by model comparison between numerous fixed (prespecified) models. So it is also important to remove very unlikely models from the starting options.</p>
<p>Model selection is typically done using AIC (<span class="math inline">\(AIC = 2k - 2l\)</span>) where <span class="math inline">\(k\)</span> (complexity) is number of parameters and <span class="math inline">\(l\)</span> (fit) is log-likelihood of data under model. In essence, penalised likelihood is a parsimonious compromise. Sometimes, model averaging can be used.</p>
</section>
<section id="tests-for-selection" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="tests-for-selection"><span class="header-section-number">6.5</span> Tests for selection</h2>
<p>Either Tajima’s <span class="math inline">\(D\)</span>, or codon-based metrics like <span class="math inline">\(\omega = dN/dS\)</span> (non-synonymous substitution per non-synonymous site vs synonymous), can be used to infer selection for new variation, i.e., positive selection in the population. People involved include Nielsen, Yang.</p>
<p>Neutrally, there should be more N than S (first two alleles of codon vs only third). But there are methods to standardise this to 1, simplifying the other cases. Purifying selection is indicated by <span class="math inline">\(\omega \lt 1\)</span>, meaning most changes in alleles are being selected against. Diversifying (positive) selection is indicated by <span class="math inline">\(\omega \gt 1\)</span>, meaning the system wants actual changes to proteins, not benign changes. In this case (if <span class="math inline">\(\omega \gt 1\)</span>) we can further use Bayesian analysis to see into which class codons go.</p>
</section>
</section>
<section id="genetic-basis-of-speciation-i-how-many-genes" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Genetic basis of speciation I: How many genes?</h1>
<p><strong>2025-10-02</strong></p>
<p>What we’ve seen so far could have been gauged based on only distributions, crosses, etc. Nowadays much more based on genomics, and at much greater scale than possible before. Some questions include:</p>
<ul>
<li>How many genes &amp; changes (few genes of large effect)?</li>
<li>Structural or regulatory?</li>
<li><em>cis</em> (close) or <em>trans</em> (far away across genome)?</li>
<li>Special role for inversions? (as a protective measure against recombination for divergent genes)</li>
</ul>
<section id="darwinian-gradualism-many-genes-of-small-effect" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="darwinian-gradualism-many-genes-of-small-effect"><span class="header-section-number">7.1</span> Darwinian gradualism | many genes of small effect</h2>
<blockquote class="blockquote">
<p>“many slight differences … afford materials for natural selection”</p>
<p>— Darwin</p>
</blockquote>
<p>Particularly favoured by biometricians who had agricultural backgrounds, and were working to improve crop growth, size, etc. <strong>Douglas Falconer</strong> was influential, with his book <em>Introduction to Quantitative Genetics</em>.</p>
</section>
<section id="mendelian-saltationism-few-genes-of-large-effect" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="mendelian-saltationism-few-genes-of-large-effect"><span class="header-section-number">7.2</span> Mendelian saltationism | few genes of large effect</h2>
<blockquote class="blockquote">
<p>“Nature does make jumps now and then”</p>
<p>— TH Huxley</p>
</blockquote>
<p>This was particularly favoured by Mendelian geneticists, who were used to looking at systems and data of big, diagnosable traits.</p>
<p>At the extreme, <strong>Waddington’s “hopeful monsters”</strong> placed faith that random aberrant (seemingly non-beneficial) mutations would somehow turn uniquely useful. e.g., Diaz et al 2020, chameleon and gecko embryo development, two-digit could have been due to single mutation as seen in the gecko.</p>
</section>
<section id="modern-synthesis" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="modern-synthesis"><span class="header-section-number">7.3</span> Modern Synthesis</h2>
<p>The two views formed a divide until the start of the 20th century, when the Modern Synthesis actually brought about reconciliation of Mendelian inheritance and Darwinian natural selection. This viewed evolution as being based on Mendelian genes, but involving <em>many alleles of small effect</em>. Thus, there was a lean towards gradualism.</p>
<p>There were several people involved in this:</p>
<ul>
<li>Julian Huxley (grandson of TH; brother Aldous; half-brother Andrew, action potentials): invented “cline”, co-founded WWF, UNESCO</li>
<li>Fisher (quantitative genetics; farm data)</li>
<li>Wright (effect pop. size; farm data)</li>
<li>Mayr (allopatric speciation)</li>
<li>Dobzhansky (<em>Drosophila</em>)</li>
<li>Stebbins (plants)</li>
<li>GG Simpson (fossils, macro-patterns)</li>
</ul>
<p><strong>Evidence</strong></p>
<p>Much of the evidence came from farm systems, and was good (held up) in crops, farm animals, etc. But people believed this without looking at wild populations. And this is also linked to the trend at the time of turning a blind eye to true ecology.</p>
<p>But as more and more actual evidence started coming in, the true picture seemed different. Orr &amp; Coyne (1992) looked at many studies and how many genes involved in adaptations. Gavrilets used theoretical models of evolution of RI and showed that fewer genes were most favoured. Yeaman &amp; Whitlock (2011) found support for fewer genes of large effect, and that genes of smaller effect tend to cluster in genome over time.</p>
</section>
<section id="which-genes" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="which-genes"><span class="header-section-number">7.4</span> Which genes?</h2>
<p>How to identify the genes responsible for speciation?</p>
<section id="forward-genetics" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="forward-genetics"><span class="header-section-number">7.4.1</span> Forward genetics</h3>
<p>Start with trait(s) in crosses, then work down to genes.</p>
<p><strong>Association mapping</strong></p>
<p>Sequence individuals for a particular trait across many individuals, and look for non-random genetic/genomic associations, i.e., look for correlatinos between markers and the trait. Recently emerged traits will be in LD with regions around them. Can use AFLPs, microsatellites, <strong>SNPs</strong>. Can be done across the whole genome: genome wide association mapping (GWAS, like Schield et al 2024).</p>
<p><strong>Issues:</strong> If goal not medical but evolutionary, results will be <strong>biased towards genes of large effect</strong>, which will drown out the many genes of small effect. It’s also affected by the <strong>extent of genomic LD</strong>: easier to pick up some region if high LD, but low precision especially with high recombination rates.</p>
<p><strong>Quantitative Trait Loci mapping (QTL)</strong></p>
<p>Create recombinant populations by first crossing between extremes for F1 (uninformative) then doing F1-parent backcrosses. From these, correlate trait values with distributed genetic markers.</p>
<p><strong>Issues:</strong> Requires pedigree info (ideally 2 generations of crosses; inbred lines important so that extremes are certainly homozygous), marker map, phenotypes associated with genotypes, need large families of individuals. Again, it has low power to detect many loci of small effect.</p>
</section>
</section>
<section id="candidate-genes" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="candidate-genes"><span class="header-section-number">7.5</span> Candidate genes</h2>
<p>Start with knowledge, or a fundamental idea, of the genetic basis for a trait in other taxa, and apply it to the new system. e.g., MC1R for melanin in birds.</p>
<p>But this prior assumption means that this is closed in two ways: it’s not open to all variation out there, and it’s also close-ended because there is no further path of questioning to follow.</p>
</section>
<section id="reverse-genetics" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="reverse-genetics"><span class="header-section-number">7.6</span> Reverse genetics</h2>
<p>Start with genomic data—ideally the whole genome—and look for signals of adaptive changes, then work back to see what traits they encode for. This of course has grown recently with the availability large whole genome sequences. Essentially, search the data for footprints of selection using genome scans, outlier analyses, etc.</p>
<p><strong>Outlier analysis</strong></p>
<p>Many methods: can compare allele frequency spectra (AFS), or look for signs of positive selection using <span class="math inline">\(\frac{D_N}{D_S}\)</span>.</p>
<p>Or, the popular <span class="math inline">\(F_{ST}\)</span> approach that was advanced by Wright: calculate the proportion of total genetic variation that is attributable to differences among subpopulations. Survey differentiated/candidate populations for many loci, calculate average overall <span class="math inline">\(F_{ST}\)</span>, simulate expected values based on neutral expectation for variation across loci, then identify the outliers. (<span class="math inline">\(F_{IT}\)</span> is the variation within an individual suspected population.) Wilding et al 2001 were one of the first to use this approach to find that shell structure in <em>Littorina saxatilis</em> varied according to tidal zone.</p>
<p>Issues of <span class="math inline">\(F_{ST}\)</span>: <strong>false positives</strong> in outliers that arise from many other kinds of variance including <strong>demographic history</strong>; and like QTL, it needs validation.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Poelstra et al 2014: hooded &amp; carrion crows
<ul>
<li>Reference genome, then analysis</li>
<li>Some loci (0.28%, among those most on same chromosome) showed clear separation, some abosolutely no</li>
<li>Important empirical followup: Later went back and got tissues from various parts ofbody, found substantial difference in gene expression (of that gene) o torso skin and not in any other part.</li>
</ul></li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p><em>152x sequence coverage vs resequencing difference?</em></p>
</div>
</div>
</section>
</section>
<section id="discussion-genetic-basis-of-speciation-i" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Genetic basis of speciation I</h1>
<p><strong>2025-10-07</strong></p>
<ul>
<li>Snails have so many sites of small differentiation because low gene flow (very low dispersal).</li>
<li>They are ovoviviparous</li>
<li>Scaffolds are fragments of genome that you make using pipeline. If same data and same pipeline, should end up with pretty similar scaffolds.</li>
</ul>
</section>
<section id="genetic-basis-of-speciation-ii-which-genes" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Genetic basis of speciation II: Which genes?</h1>
<p><strong>2025-10-09</strong></p>
<section id="structural-vs-regulatory-sequences" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="structural-vs-regulatory-sequences"><span class="header-section-number">8.1</span> Structural vs regulatory sequences</h2>
<p>Structural gene sequences control what is being made, whereas regulatory sequences control when and how much is being made. Most regulation happens at the level of transcription, but some at translation via attachment of siRNAs (small interfering; becomes double stranded when attach to mRNA so no translation).</p>
<p>In crows, colour divergence was not always necessarily due to differences in amino acids but rather in coding regions like introns. Do regulatory changes underlie qualitatively different types of phenotypic change?</p>
</section>
<section id="cis-vs-trans-mechanism-of-regulation" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="cis-vs-trans-mechanism-of-regulation"><span class="header-section-number">8.2</span> <em>cis</em> vs <em>trans</em> | mechanism of regulation</h2>
<p>Regulation of transcription fundamentally occurs due to sequence-specific binding (close to protein-coding region, so (<em>cis-</em>)) of transcription factors (<em>trans-</em>, far across).</p>
<p>cis-regulatory elements are regions of non-coding DNA that regulate the transcription of nearby genes; they act at the intra-molecular level via binding sequences.</p>
<p>trans-regulatory elements are proteins that regulate the transcription of distant genes; they act at mostly the inter-molecular level, on many genes (not 1:1). aka <em>transcription factors</em>.</p>
</section>
<section id="cis-regulatory-changes-structural-changes" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="cis-regulatory-changes-structural-changes"><span class="header-section-number">8.3</span> cis-regulatory changes &gt; structural changes</h2>
<blockquote class="blockquote">
<p>“regulatory evolution is <em>the</em> creative force underlying morphological diversity across the evolutionary spectrum, from variation within species to body plans”</p>
<p>— Sean Carroll</p>
</blockquote>
<p>Sean Carroll: UoWisconsin, HHMI funding (millions over 7 years, renewable). Showed evidence for convergent evolution of CRE for the same gene in multiple <em>Drosophila</em> species. Also wrote popular books.</p>
<p>Proponents: Britten &amp; Davidson, Wray, Valentine. Rebuttal: Hoekstra &amp; Coyne (2007).</p>
<p>Sean Carroll’s argument is founded on several factors:</p>
<ul>
<li>Mosaic pleiotropy: most proteins regulating development participate in multiple independent developmental processes.</li>
<li>Ancestral genetic complexity: morphologically disparate and long-diverged animal taxa share similar toolkits of body-building and -patterning genes.</li>
<li>Functional equivalence of distant homologs: many animal toolkit proteins exhibit functional equivalency (redundancy) in vivo when substituted for one another.</li>
<li>Deep homology: formation and differentiation of many structures are governed by similar sets of genes and regulatory circuits.</li>
<li>Infrequent toolkit gene duplication</li>
<li>Heterotopy: changes in spatial regulation of toolkit genes and the genes they regulate are associated with morphological divergence.</li>
<li>Modularity of CRE: can have multiple CRE for same genes controlling different aspects of it, like differential expression in different tissues/body parts. (<em>But counter argument by Hoekstra et al.&nbsp;is that duplication can give the same result by copying both CRE and gene.</em>)</li>
<li>Vast regulatory networks of transcription factors: individual regulatory proteins (transcription factors) control scores to hundreds of target gene CREs; so changes to TF are “catastrophic”.</li>
</ul>
</section>
<section id="in-defence-of-structural-changes" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="in-defence-of-structural-changes"><span class="header-section-number">8.4</span> In defence of structural changes</h2>
<p>Hoekstra &amp; Coyne (2007): morphological changes are actually not so special. Gene duplications can accomplish many of cis-regulatory genes’ special claims (we now know gene duplication rates are almost similar to point mutation rates). And some “regulatory” changes are actually structural: transcription factors include some of fastest evolving human genes (?).</p>
<p>At the time, there was also more evidence of protein changes than CR changes, but later we have seen that both factors are important. Interestingly, both Carroll and Hoekstra ended up publishing later concluding the very opposite of their own earlier POVs.</p>
<p>Much of this debate has placed a strong focus on characters/traits, and is not necessarily tied to speciation in closely related species.</p>
</section>
</section>
<section id="discussion-genetic-basis-of-speciation-ii-gene-regulation" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Genetic basis of speciation II (Gene regulation)</h1>
<p><strong>2025-10-21</strong></p>
<ul>
<li>Fundamental of -omics approach: look at everything (genomes or transcriptomes), let data show you who’s interesting</li>
<li>Genome size related to cell size, so in flying birds for metabolic reasons cells have to be small and therefore genomes small. Same in bats versus other mammals.</li>
</ul>
</section>
<section id="genomic-architecture" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Genomic architecture</h1>
<p>aka, Islands of Discord</p>
<p><strong>2025-10-23</strong></p>
<section id="genomic-islands-of-speciation-differentiation" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="genomic-islands-of-speciation-differentiation"><span class="header-section-number">9.1</span> Genomic islands of speciation (differentiation)</h2>
<p><strong>Felsenstein’s paradox</strong> made him “[skeptical] towards Santa Rosalia” (vs Hutchinson): recombination brings good genes together, but makes it hard for them to <em>stay</em> together!</p>
<p>Genomic islands of differentiation are relatively small regions of the genome that have elevated divergence, surrounded by regions of low/no divergence. This concept arose (in “speciation” flavour) due to multiple key influences around the same time, ~ 20 years ago:</p>
<ul>
<li>Rebirth of reinforcement</li>
<li>Genic speciation, Chung-I Wu (versed Coyne): shifting emphasis from isolation at whole genome to individual genes</li>
<li>Isolation-with-migration model (Nielsen, Wakeley, Hey): shift from coalescent approaches to speciation with gene flow</li>
<li>Rieseberg et al 1996. <em>Helianthus anomalus</em> is a dune specialist. Non-random hybrid composition between two parental lineages, even in lab crosses, very similar to that in wild hybrid. “Interactions between coadapted parental species’ genes constrain the genomic composition of hybrid species.”</li>
<li>Turner et al 2005. <em>Anopheles gambiae</em>: Three regions (&lt; 2.8 Mb) contain most differentiation, and these speciation islands remain differentiated despite considerable gene flow (therefore responsible for RI).</li>
</ul>
<p>By 2010–15, speciation with gene flow was already dogmatic. GIDs are thus in regions of <strong>low recombination</strong>, and <strong>co-localise with species-defining traits</strong> (e.g., hood in Hooded Crow). There is also <strong>LD among unlinked islands</strong> (e.g., capuchino seedeaters), and island (gene) trees match species trees (<strong>gene tree–species tree concordance</strong>). Further, the islands <strong>grow over time</strong> (Via 2012) because recombination fragment size is larger than individual genes, and some genes/loci hitchhike along with loci under selection.</p>
<p>GIDs can involve inversions (rearrangement/flipping of gene order in chromatin packing; these areas simply do not get recombined, stay the same, due to the differing order; Noor et al 2001 inc. Jane Reiland: QTLs also associated with regions of inversion) and structural genomic variants (beyond just SNPs) which are often associated with adaptations, clines, and speciation (Hooper &amp; Price 2015).</p>
</section>
<section id="disputed-islands" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="disputed-islands"><span class="header-section-number">9.2</span> Disputed islands</h2>
<p>Turner &amp; Hahn 2010: what is “divergence”? Most approaches focused on relative divergence, comparing between vs within (e.g., <span class="math inline">\(F_{ST}\)</span> compares sub with tot). Markers with high variation within populations are less able to show between-variation (max <span class="math inline">\(F_{ST}\)</span> will be low when <span class="math inline">\(H_{S}\)</span>, i.e., recombination, high). Conversely, low recombination within population will drive up <span class="math inline">\(F_{ST}\)</span>.</p>
<p>On the other hand, absolute divergence (e.g., <span class="math inline">\(d_{xy}\)</span>) measures the average number of pairwise differences between sequences from two species, and is not sensitive to within-population variation. GIS should have both relative and absolute divergence high, so shift to GID.</p>
<p><em>Genomic islands</em> of <em>speciation or genomic islands</em> and <em>speciation?</em></p>
<p>Cruickshank &amp; Hahn 2014 showed that absolute divergence doesn’t keep up, supporting the divergence-after-speciation model. Thus, the view changed:</p>
<ul>
<li><del>In regions of low recombination</del></li>
<li><del>co-localise with species-defining traits</del> <span class="math inline">\(\rightarrow\)</span> traits arose post-speciation</li>
<li><del>LD among unlinked islands</del> <span class="math inline">\(\rightarrow\)</span> impossible to maintain</li>
<li><del>gene tree–species tree concordance</del> <span class="math inline">\(\rightarrow\)</span> selection speeds lineage sorting</li>
</ul>
<p>Guerrero &amp; Hahn 2017: Speciation only responsible for sieving (sorting) process. Divergence itself ancestral (polymorphism in population). Ancestral balanced polymorphisms are sorted unequally between descendant lineages (balanced: both coexist in population).</p>
</section>
</section>
<section id="discussion-genomic-architecture" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Genomic architecture</h1>
<p><strong>2025-10-28</strong></p>
<ul>
<li>Parallel speciation: Speciation version of convergent evolution. Parallel evolution assumes more similar starting points, is a subset of convergent evolution.</li>
<li>Type I inversions: divergent/balancing selection. Type II: good things on both inversions, so best is to have heterozygous (e.g., trapped along with good stuff might be bad, and some bad effects may be covered by having heterokaryotype). (Zygote: single locus; karyotype: can be set of loci, structural variants.)</li>
</ul>
</section>
<section id="hybridisation-and-introgression" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Hybridisation and introgression</h1>
<p><strong>2025-10-30</strong></p>
<p>Hybridisation is reproduction between members of genetically differentiated populations that yields offspring of mixed ancestry. e.g., Carrion &amp; Hooded Crows. It is often initiated by factors like range change, habitat shifts/mosaics, disturbance.</p>
<p>The <em>outcomes</em> of hybridisation include fusion, hybrid speciation, balance, and introgression.</p>
<section id="fusion" class="level2" data-number="10.1">
<h2 data-number="10.1" class="anchored" data-anchor-id="fusion"><span class="header-section-number">10.1</span> Fusion</h2>
<p>This is the reversal of speciation, the wholesale collapse of a distinct entity which continues to be more homogenised.</p>
<p>A good example is Common &amp; Chihuahuan Raven. Chi. originally split from Common, then one fused back with Common forming the California type, while the other remains Chi. So the California lineage is considered to have fused (subsumed by Common).</p>
</section>
<section id="hybrid-speciation" class="level2" data-number="10.2">
<h2 data-number="10.2" class="anchored" data-anchor-id="hybrid-speciation"><span class="header-section-number">10.2</span> Hybrid speciation</h2>
<p>This forms a transgressive variant, a third group. Allopolyploid speciation is driven by the different numbers of chromosomes, and is common in angiosperms, ferns, <em>Acropora</em>. Homoploid hybrid speciation happens with no ploidy changes, and more common in animals , e.g., <em>Cottus</em> fish, sparrows (House, Italian, Spanish), snub-nosed monkey.</p>
<p>Often happens due to transgressive phenotypes: hybridisation generates fitness-related phenotypes that lie outside the parental distribution.</p>
</section>
<section id="balance" class="level2" data-number="10.3">
<h2 data-number="10.3" class="anchored" data-anchor-id="balance"><span class="header-section-number">10.3</span> Balance</h2>
<p>Forms hybrid zones. Generally involves habitat mosaics, and is common in mice, mussels, crickets.</p>
<p><strong>Tension zones</strong> are an interesting case where <em>dispersal</em> into zone fights with <em>selection</em> against crossing. (See Fire-bellied Toad, Nick Barton.) Measure strength of tension by <span class="math inline">\(\theta = \frac{S}{R}\)</span> (ratio of effective selection and pairwise recombination rate).</p>
<p>Hybrid zones tend to be drawn to areas of low population density, and show spatial coupling of genotypic and phenotypic clines enhanced by reduced recombination, epistasis among barrier loci, moving clines to low fitness landscape (?).</p>
<p>Brumfield et al 2001: clines in manakins do not line up with hybrid zones. The transition in certain markers are at some different geographical location than the actual hybrid zone (where crucial trait also transitions), so looking at just that would suggest introgression.</p>
</section>
<section id="introgression" class="level2" data-number="10.4">
<h2 data-number="10.4" class="anchored" data-anchor-id="introgression"><span class="header-section-number">10.4</span> Introgression</h2>
<p>The seeping of a small set of genes, that allow adaptations to jump the species barrier (cf fusion). Something that was originally in one species ends up in another, and this is therefore kind of an exception because it manages to escape the forces acting to keep it within the hybrid zone.</p>
<p>But this allows hybridisation to act as the important source of <em>adaptive variation</em>, instead of mutations which are rare. Examples include Grants in Galapagos finches; high altitude adaptation in Tibetans (adaptive introgression with denisovans); <em>Heliconius</em> aposematism: same morph in sympatry, different in allopatry.</p>
<p><strong>ABBA-BABA test:</strong> Patterson’s <span class="math inline">\(D =  \frac{ABBA - BABA}{ABBA + BABA}\)</span>. Tests for introgression (<span class="math inline">\(ABBA &gt; BABA\)</span>) vs incomplete lineage sorting (<span class="math inline">\(ABBA = BABA\)</span> cos random). Need four groups, and large genome if not whole genome.</p>
<p><strong>Migrant tracts:</strong> chromosome segments that show migrant ancestry. Length distribution of chromosome segments will change over time due to recombination. Archaic ancestor SNPs will become smaller/rarer farther away from introgression event. Can compare increasing vs decreasing vs constant introgression. Can also identify private alleles (only found in one population): if due to <em>de novo</em> mutations, random distribution, but if past introgression, expect longer runs of identity in SNPs.</p>
<p><em>Is introgression the new allopatry?</em></p>
</section>
</section>
<section id="discussion-introgression" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Introgression</h1>
<p><strong>2025-11-04</strong></p>
<ul>
<li>Looks like Jensen et al 2024 did not start out with hypothesis/prediction, instead got the data and framed the story around it.</li>
<li>PSMC: only need one individual to recreate the population history</li>
</ul>
</section>
<section id="old-variation-is-the-new-thing" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Old variation is the new thing</h1>
<p><strong>2025-11-06</strong></p>
<section id="de-novo-variation" class="level2" data-number="11.1">
<h2 data-number="11.1" class="anchored" data-anchor-id="de-novo-variation"><span class="header-section-number">11.1</span> <em>de novo</em> variation</h2>
<p>DMI can arise through point mutations, duplication, scrambling, retroposition, mobile elements, lateral gene transfer (viruses, parasites), fission and fusion, chimeric genes (origins in separate genes).</p>
<p>But this is a long waiting game, most mutations are deleterious and start at low frequencies (<span class="math inline">\(\frac{1}{2} N_{e})\)</span>), and small effect mutations are genomically diffuse.</p>
<p><em>Geospiza difficilis</em> has few genes with large effect on beak size, thus an ecological impact along with an easily identifiable morphological feature. When looking at the overall species tree, the genetic basis for species difference is older than the basis for species split itself. The current sister species have blunt or pointed beaks, so introgression with blunt type introduced the blunt beak in new lineages.</p>
</section>
<section id="standing-variation" class="level2" data-number="11.2">
<h2 data-number="11.2" class="anchored" data-anchor-id="standing-variation"><span class="header-section-number">11.2</span> Standing variation</h2>
<p>The diversity of genetic variants present in a population (can include <em>de novo</em> mutation).</p>
<p><strong>Ancestral variation:</strong> genetic variation present in common ancestor of populations/species being compared. Forms the substrate for rapid divergence, and explains why speciation alleles often predate species divergence. No waiting, pre-screened for deleterious alleles, high starting frequency, small effect alleles clustered (so enriched for large-effect haplotypes/islands), co-adapted genes clustered (LD). Many sources: mutation, large <span class="math inline">\(N_{e}\)</span>, balanced polymorphisms (broad geo/ecological range species, e.g., balanced alleles for two different habitats), introgression (admixture variation).</p>
</section>
<section id="combinatorial-view-of-speciation" class="level2" data-number="11.3">
<h2 data-number="11.3" class="anchored" data-anchor-id="combinatorial-view-of-speciation"><span class="header-section-number">11.3</span> Combinatorial view of speciation</h2>
<p><em>de novo</em> vs standing variation <span class="math inline">\(\sim\)</span> gene flow vs sieve speciation.</p>
<p><em>Old hybridisation creates variation that can be sorted into new species.</em></p>
<p>Seehausen 2005: beginning of speciation event is not the origin but the reassembly of several old variants into novel combinations. <strong>Hybrid swarm:</strong> hybridisation upon invasion of new habitat producing lots of heritable variation for ecological traits. <strong>Syngameon:</strong> a complex of genetically weakly but ecologically highly distinctive species capable of exchanging genetic material (in context of radiations).</p>
<p><strong>Implications</strong></p>
<ul>
<li>Decoupling of mutation and speciation rates</li>
<li>Facilitates evolution of LD in face of gene flow</li>
<li>These two together: no wait when ecological opportunity arises</li>
</ul>
<p>Things left to do:</p>
<ul>
<li>Rapid (inc. non-adaptive) radiations vs depauperate sister taxa</li>
<li>Allele splits vs species splits</li>
<li>Compare LD among species</li>
<li>Locus size effects for old vs new variants</li>
<li>Revisit old data (e.g., incongruent gene trees)</li>
</ul>
</section>
</section>
<section id="discussion-old-variation" class="level1 unnumbered">
<h1 class="unnumbered">Discussion: Old variation</h1>
<p><strong>2025-11-11</strong></p>
<ul>
<li>Mitochondrial DNA has way more drift for same eff pop size than nuclear, which is why you usually see discordance.</li>
</ul>
</section>
<section id="microbes-microbiomes-and-speciation" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Microbes, microbiomes and speciation</h1>
<p><strong>2025-11-13</strong></p>
<section id="species" class="level2" data-number="12.1">
<h2 data-number="12.1" class="anchored" data-anchor-id="species"><span class="header-section-number">12.1</span> Species</h2>
<p>First of all, only 1-2% of bacteria are culturable, so biased view (<strong>unculturable paradigm</strong>). Heuristic for bacterial species concept is 3% divergence at small-ribosomal subunit (SSU) rRNA.</p>
<p>Why SSU?</p>
<ul>
<li>Encoded by nucleus and organelles (e.g, 16S)</li>
<li>Slow and fast (hour-hands, minute-hands)</li>
<li>Conserved structure</li>
<li>Practicalities</li>
<li>Ancient and essential (ribosome)</li>
<li>Highly integrated (&gt;100 co-evolved cellular RNAs and proteins). Should also minimise LGT</li>
<li>Lateral/Horizontal Gene Transfer: Movement of genetic material between organisms other than by vertical (parent-to-offspring) transmission.</li>
</ul>
<p>Problems with SSU rRNA:</p>
<ul>
<li>Discordance with DNA hybridisation: overall rates of DNA similarity (reannealing) did not match with those of SSU rRNA</li>
<li>Ecological differentiation within “species”: 3% rule of thumb doesn’t capture full picture.</li>
<li>Descriptive, not a species <em>concept</em></li>
</ul>
<p>Reality of species, and how recombination works, is different in bacteria. There is no meiosis, so recombination is rare and not every single generation. On the other hand, recombination can happen with a wide array of organisms (promiscuous), and is highly localised, adding/dropping individual genes at very small levels. There can be transfer of non-homologous loci, and <strong>exogenous DNA is a major raw material for adaptation</strong>.</p>
<p>The basis of bacterial species cohesion involves extinction/recolonisation (metapopulation model, bottlenecks) and stabilising selection on ecotypes.</p>
</section>
<section id="communities" class="level2" data-number="12.2">
<h2 data-number="12.2" class="anchored" data-anchor-id="communities"><span class="header-section-number">12.2</span> Communities</h2>
<p>Are there bacterial communities? From different habitats? On/in different hosts? How tight are associations between species or between species and environment?</p>
<p>Yes, habitat-wise. Yes, host-specific. Tight association with host life cycle (transfer of symbionts/microbiota).</p>
<p><strong>Microbiome:</strong> Microbes that live on and inside a defined habitat (e.g., an organism’s body). Human body has 10x prokaryotic cells than eukaryotic. e.g., <em>C. difficile</em> is natural part of microbiome, but in hospitals the antibiotics around affect other bacteria that hold it down in normal condition. Fecal transplants help with this more than antibiotics.</p>
</section>
<section id="holobiont" class="level2" data-number="12.3">
<h2 data-number="12.3" class="anchored" data-anchor-id="holobiont"><span class="header-section-number">12.3</span> Holobiont</h2>
<p>A host and its microbiota. Multicellular eukaryotes are not and never have been autonomous organisms.</p>
<p><strong>Hologenome:</strong> the complete genetic content of the host genome, its organelles, and its microbiome. Host’s nuclear genome inherited <em>a la</em> Mendel, but microbiome is Lamarckian and often uniparental! So, sources of holobiont variation are:</p>
<ul>
<li>SNPs</li>
<li>Recombination</li>
<li>Gene loss and duplication</li>
<li><strong>Microbe loss or amplification</strong></li>
<li><strong>Horizontal gene transfer</strong>: Red alga seaweed <em>Porphyra</em>, genes that allow humans to digest, usually only found on bacteria that live on the plant, were transferred via HGT to people in Japan, especially mother-child (Hehemann et al 2010).</li>
</ul>
<p>Implication for evolution is that microbes in microbiome can be equated with genes in genome. Role of host-associated microbes in speciation of host can be tested using microbe-free (antibiotic) versions of same old experiments. e.g., Brucker &amp; Bordenstein 2013: <em>Nasonia</em> wasps and microbiome affecting viability.</p>
<p><strong>Phylosymbiosis:</strong> microbial community similarity that mirrors the phylogeny of their host (Lim &amp; Bordenstein 2020). Dunaj et al 2020 showed phylosymbiosis in ovaries and silk glands but not fat/midgut.</p>
<p><strong>Cophylogeny:</strong> agreement of evolutionary histories of symbionts and their hosts (Mark Hafner LSU; not bacteria, but lice and the host gophers).</p>
</section>
</section>
<section id="macroevolution" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Macroevolution</h1>
<p><strong>2025-11-20</strong></p>
<p>Evolution above the species level, studying:</p>
<ul>
<li>Patterns of diversity over time
<ul>
<li>Mode and tempo of evolution</li>
<li>Speciation <em>and extinction</em></li>
</ul></li>
<li>Drivers of patterns of diversity over time
<ul>
<li>Key innovations (tied to rates)</li>
<li>Ecological/geographical circumstances</li>
</ul></li>
<li>Species selection
<ul>
<li><em>Can natural selection occur at levels higher than species?</em></li>
<li>Diversification of clades caused by heritable species-level traits</li>
<li>Rabosky &amp; McCune 2010?</li>
<li>Problem was need trait that’s species-level, not individual-level. e.g., polymorphism, <em>range size</em></li>
</ul></li>
</ul>
<section id="fossil-record" class="level2" data-number="13.1">
<h2 data-number="13.1" class="anchored" data-anchor-id="fossil-record"><span class="header-section-number">13.1</span> Fossil record</h2>
<p>Can help:</p>
<ul>
<li>Calibrate molecular clock</li>
<li>Identify transitional forms</li>
<li>Unveil morphological history</li>
<li>Unveil ecological history</li>
<li>Estimate extinction rates
<ul>
<li>Mass extinctions are only 4% of all extinctions; others are all <em>background extinctions</em></li>
<li>Terminal Ordovician (444 Ma), Late Devonian (360 Ma), End Permian (251 Ma), End Triassic (200 Ma), Cretaceous-Paleogene (65.5 Ma)</li>
<li>Biotic replacement (e.g., ecological): brachiopods vs bivalves</li>
</ul></li>
<li>Estimate “tempo and mode” of speciation
<ul>
<li>Punctuated equilibrium (SJ Gould, Eldredge), aka “evolution by jerks”
<ul>
<li>Most morphological change accumulates over short time, then stays quiet for long time (vs phyletic gradualism)</li>
<li>(Gould influenced by Marxist ideas)</li>
<li>Main counter: this pattern likely due to incomplete fossil record</li>
<li>But examples of adaptive radiation, etc. suggest this may not be all impossible</li>
</ul></li>
<li>So the more remarkable pattern is sometimes patterns of <em>stasis</em> (at least morphological) in fossil records, despite rapid change elsewhere. e.g.&nbsp;in Louisiana, horseshoe crab, gar: genetic evolution rates similar between horseshoe crab and hermit crab, so must be something else about organisation of genome.</li>
</ul></li>
</ul>
<p>However, ILS throws off fossil calibrations. Also, if fossils are old and you’re looking at recent events, bias (and converse).</p>
</section>
<section id="phylogenetic-trees" class="level2" data-number="13.2">
<h2 data-number="13.2" class="anchored" data-anchor-id="phylogenetic-trees"><span class="header-section-number">13.2</span> Phylogenetic trees</h2>
<p>Can study:</p>
<ul>
<li>Changes in no. of surviving lineages over time (net diversification rate = speciation rate - extinction rate)
<ul>
<li>Speciation rate: Yule process (birth/speciation only), generates distribution of clade ages and sizes</li>
<li>Extinction rate: contingent on sampling completeness, branch length estimation, constancy rates</li>
<li>In general, our <strong>methods are really good at inferring topology (branch order), but not good at nailing down branch lengths</strong></li>
</ul></li>
<li>Detecting diversification patterns
<ul>
<li>Sister clade comparison. e.g., Farrell 1998: beetles and angiosperm associations.</li>
<li>Lineage-through time (LTT) plots
<ul>
<li>How many lineages exist at a specific time, over a period of time (e.g., Glor 2010)</li>
<li>Test: slow constant species accumulation vs adaptive radiation</li>
<li>Bromeliads: C3-C4 vs tankless–tank-forming</li>
</ul></li>
<li>Adaptive radiation: short early branches and long later branches
<ul>
<li>Antarctic icefish: anti-freeze glycoproteins (hyperdiver) prevent growth of ice crystals</li>
</ul></li>
<li>BAMM: Bayesian Analysis of Macroevolutionary Mixtures (Rabosky 2014). Like cluster analysis, but this identifies how many different groups of differential diversification rates exist in your tree.
<ul>
<li>But likelihood ignores rate shifts on extinct lineages, biases estimates of extinction probabilities. CPP prior used for distribution of rate shifts over tree are too sensitive. (Moore et al 2016).</li>
</ul></li>
</ul></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>