<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Reading notes – BIOL 7117: Speciation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6b3bef205c96c4f74d1e1b8d3d254fec.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIOL 7117: Speciation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./class-notes.html"> 
<span class="menu-text">Class notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./readings.html" aria-current="page"> 
<span class="menu-text">Reading notes</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#mallet-2013" id="toc-mallet-2013" class="nav-link active" data-scroll-target="#mallet-2013"><span class="header-section-number">1</span> Mallet 2013</a></li>
  <li><a href="#foote-2018" id="toc-foote-2018" class="nav-link" data-scroll-target="#foote-2018"><span class="header-section-number">2</span> Foote 2018</a></li>
  <li><a href="#discussion-geography-of-speciation" id="toc-discussion-geography-of-speciation" class="nav-link" data-scroll-target="#discussion-geography-of-speciation"><span class="header-section-number">3</span> Discussion: Geography of speciation</a>
  <ul class="collapse">
  <li><a href="#coathup-et-al-2024" id="toc-coathup-et-al-2024" class="nav-link" data-scroll-target="#coathup-et-al-2024"><span class="header-section-number">3.1</span> Coathup et al 2024</a></li>
  <li><a href="#dean-et-al-2019" id="toc-dean-et-al-2019" class="nav-link" data-scroll-target="#dean-et-al-2019"><span class="header-section-number">3.2</span> Dean et al 2019</a></li>
  </ul></li>
  <li><a href="#rundle-nosil-2005" id="toc-rundle-nosil-2005" class="nav-link" data-scroll-target="#rundle-nosil-2005"><span class="header-section-number">4</span> Rundle &amp; Nosil 2005</a>
  <ul class="collapse">
  <li><a href="#ecological-causes-of-divergent-selection" id="toc-ecological-causes-of-divergent-selection" class="nav-link" data-scroll-target="#ecological-causes-of-divergent-selection"><span class="header-section-number">4.1</span> Ecological causes of divergent selection</a></li>
  </ul></li>
  <li><a href="#discussion-pre-zygotic-isolation" id="toc-discussion-pre-zygotic-isolation" class="nav-link" data-scroll-target="#discussion-pre-zygotic-isolation"><span class="header-section-number">5</span> Discussion: Pre-zygotic isolation</a>
  <ul class="collapse">
  <li><a href="#zhang-et-al-2024" id="toc-zhang-et-al-2024" class="nav-link" data-scroll-target="#zhang-et-al-2024"><span class="header-section-number">5.1</span> Zhang et al 2024</a></li>
  <li><a href="#schield-et-al-2024" id="toc-schield-et-al-2024" class="nav-link" data-scroll-target="#schield-et-al-2024"><span class="header-section-number">5.2</span> Schield et al 2024</a></li>
  </ul></li>
  <li><a href="#discussion-reinforcement" id="toc-discussion-reinforcement" class="nav-link" data-scroll-target="#discussion-reinforcement"><span class="header-section-number">6</span> Discussion: Reinforcement</a>
  <ul class="collapse">
  <li><a href="#bemmels-et-al-2021" id="toc-bemmels-et-al-2021" class="nav-link" data-scroll-target="#bemmels-et-al-2021"><span class="header-section-number">6.1</span> Bemmels et al 2021</a></li>
  <li><a href="#levitan-hao-2025" id="toc-levitan-hao-2025" class="nav-link" data-scroll-target="#levitan-hao-2025"><span class="header-section-number">6.2</span> Levitan &amp; Hao 2025</a></li>
  </ul></li>
  <li><a href="#discussion-genetic-basis-of-speciation-i" id="toc-discussion-genetic-basis-of-speciation-i" class="nav-link" data-scroll-target="#discussion-genetic-basis-of-speciation-i"><span class="header-section-number">7</span> Discussion: Genetic basis of speciation I</a>
  <ul class="collapse">
  <li><a href="#turbek-et-al-2021" id="toc-turbek-et-al-2021" class="nav-link" data-scroll-target="#turbek-et-al-2021"><span class="header-section-number">7.1</span> Turbek et al 2021</a></li>
  <li><a href="#raffini-et-al-2025" id="toc-raffini-et-al-2025" class="nav-link" data-scroll-target="#raffini-et-al-2025"><span class="header-section-number">7.2</span> Raffini et al 2025</a></li>
  </ul></li>
  <li><a href="#discussion-genetic-basis-of-speciation-ii-gene-regulation" id="toc-discussion-genetic-basis-of-speciation-ii-gene-regulation" class="nav-link" data-scroll-target="#discussion-genetic-basis-of-speciation-ii-gene-regulation"><span class="header-section-number">8</span> Discussion: Genetic basis of speciation II: Gene regulation</a>
  <ul class="collapse">
  <li><a href="#chen-et-al-2024" id="toc-chen-et-al-2024" class="nav-link" data-scroll-target="#chen-et-al-2024"><span class="header-section-number">8.1</span> Chen et al 2024</a></li>
  <li><a href="#yusuf-et-al-2025" id="toc-yusuf-et-al-2025" class="nav-link" data-scroll-target="#yusuf-et-al-2025"><span class="header-section-number">8.2</span> Yusuf et al 2025</a></li>
  </ul></li>
  <li><a href="#discussion-genome-architecture" id="toc-discussion-genome-architecture" class="nav-link" data-scroll-target="#discussion-genome-architecture"><span class="header-section-number">9</span> Discussion: Genome architecture</a>
  <ul class="collapse">
  <li><a href="#meyer-et-al-2024" id="toc-meyer-et-al-2024" class="nav-link" data-scroll-target="#meyer-et-al-2024"><span class="header-section-number">9.1</span> Meyer et al 2024</a></li>
  <li><a href="#huang-et-al-2025" id="toc-huang-et-al-2025" class="nav-link" data-scroll-target="#huang-et-al-2025"><span class="header-section-number">9.2</span> Huang et al 2025</a></li>
  </ul></li>
  <li><a href="#discussion-introgression" id="toc-discussion-introgression" class="nav-link" data-scroll-target="#discussion-introgression"><span class="header-section-number">10</span> Discussion: Introgression</a>
  <ul class="collapse">
  <li><a href="#jensen-et-al-2024" id="toc-jensen-et-al-2024" class="nav-link" data-scroll-target="#jensen-et-al-2024"><span class="header-section-number">10.1</span> Jensen et al 2024</a></li>
  <li><a href="#jorquera-et-al-2025" id="toc-jorquera-et-al-2025" class="nav-link" data-scroll-target="#jorquera-et-al-2025"><span class="header-section-number">10.2</span> Jorquera et al 2025</a></li>
  </ul></li>
  <li><a href="#discussion-old-variation" id="toc-discussion-old-variation" class="nav-link" data-scroll-target="#discussion-old-variation"><span class="header-section-number">11</span> Discussion: Old variation</a>
  <ul class="collapse">
  <li><a href="#meier-et-al-2023" id="toc-meier-et-al-2023" class="nav-link" data-scroll-target="#meier-et-al-2023"><span class="header-section-number">11.1</span> Meier et al 2023</a></li>
  <li><a href="#lutgen-et-al-2025" id="toc-lutgen-et-al-2025" class="nav-link" data-scroll-target="#lutgen-et-al-2025"><span class="header-section-number">11.2</span> Lutgen et al 2025</a></li>
  </ul></li>
  <li><a href="#discussion-microbes-and-microbiomes" id="toc-discussion-microbes-and-microbiomes" class="nav-link" data-scroll-target="#discussion-microbes-and-microbiomes"><span class="header-section-number">12</span> Discussion: Microbes and microbiomes</a>
  <ul class="collapse">
  <li><a href="#wang-et-al-2021" id="toc-wang-et-al-2021" class="nav-link" data-scroll-target="#wang-et-al-2021"><span class="header-section-number">12.1</span> Wang et al 2021</a></li>
  <li><a href="#baiz-et-al-2024" id="toc-baiz-et-al-2024" class="nav-link" data-scroll-target="#baiz-et-al-2024"><span class="header-section-number">12.2</span> Baiz et al 2024</a></li>
  </ul></li>
  <li><a href="#discussion-macroevolution" id="toc-discussion-macroevolution" class="nav-link" data-scroll-target="#discussion-macroevolution"><span class="header-section-number">13</span> Discussion: Macroevolution</a>
  <ul class="collapse">
  <li><a href="#heuer-et-al-2024" id="toc-heuer-et-al-2024" class="nav-link" data-scroll-target="#heuer-et-al-2024"><span class="header-section-number">13.1</span> Heuer et al 2024</a></li>
  <li><a href="#title-et-al-2024" id="toc-title-et-al-2024" class="nav-link" data-scroll-target="#title-et-al-2024"><span class="header-section-number">13.2</span> Title et al 2024</a></li>
  </ul></li>
  <li><a href="#discussion-macroevolution-1" id="toc-discussion-macroevolution-1" class="nav-link" data-scroll-target="#discussion-macroevolution-1"><span class="header-section-number">14</span> Discussion: Macroevolution</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Reading notes</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="mallet-2013" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Mallet 2013</h1>
<ul>
<li></li>
</ul>
</section>
<section id="foote-2018" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Foote 2018</h1>
<ul>
<li>Sympatric speciation does not involve physical/geographic barriers, so offers insights into role of selection (natural and sexual) in driving speciation.</li>
<li>This speciation (divergence with gene flow) is therefore rare, extreme, restrictive, and controversial.</li>
<li>Genomic methods have provided evidence of two kinds: either multiple colonisations and secondary contact, or actual sympatric divergence.</li>
<li>Author overall argues for less focus on primary vs secondary contact as driving speciation, and more on mechanistic aspects of genomic architecture.</li>
<li>Previous methods used phylogenetics to infer sympatric speciation from monophyly, but monophyly is not a good indicator of sympatric speciation (consistent with, but not exclusive to). (One flaw of Coyne &amp; Orr’s criteria.)</li>
<li>Backwards-in-time approaches are limited to identifying gene flow events arising through coalescence. They cannot distinguish between periods of spatial isolation and of spatial overlap with on gene flow.</li>
<li>In cases of actual sympatric speciation, we expect barrier loci to be clustered tightly (close together) in the genome, and at areas of low recombination, and to have a “large effect” especially affecting both reproductive isolation and ecological adaptation. That is, <em>linkage</em> or <em>pleiotropy</em>. Otherwise, ongoing gene flow (recombination, genome-wide homogenisation) would make it difficult for divergence.</li>
<li>Modern genomic methods are still biased towards disproving (and not supporting) primary sympatric speciation. Palaeogenomics is a useful tool to tackle this.</li>
<li>Author argues that focus should be placed on sympatric speciation, and away from primary vs secondary.</li>
</ul>
</section>
<section id="discussion-geography-of-speciation" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Discussion: Geography of speciation</h1>
<section id="coathup-et-al-2024" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="coathup-et-al-2024"><span class="header-section-number">3.1</span> Coathup et al 2024</h2>
<ul>
<li>Theory says sympatric speciation facilitated by magic traits which might be encoded by pleiotropic genes, but little evidence exists of sympatric speciation and empirical evidence of magic traits is rare. Authors investigated role of pleitropy in <em>Howea</em> speciation. They found multiple genes showing adaptive divergence between the sister species (sign of pleiotropy, in soil stress tolerance and flowering time).</li>
<li>Alternative to magic traits/pleiotropy driving sympatric ecological speciation is linkage disequilibrium. (i.e., one allele with multiple effects vs two close alleles not separated in recombination.) In either case, “major effect loci”.</li>
<li>Dunning et al 2016 and Osborne et al 2019 assessed gene expression and genetic divergence. They identified candidate pleiotropic speciation genes. The current study aimed to test whether these genes actually are pleiotropic loci affecting soil stress tolerance and flowering time. If so, this would be positive evidence for magic traits.</li>
<li>Method: T-DNA knockout <em>A. thaliana</em> used for high-throughput phenotyping (knockouts in orthologues of candidate speciation genes, 11 knockout mutants).
<ul>
<li>Grown under various stress emulating Lord Howe Island. Then morphology, stress responses and flowering time measured.</li>
</ul></li>
<li>Five genes potential speciation genes: knockouts in each of these showed displaced flowering times and altered soil stress response (cadmium, drought, salt). Largest effect sizes in SAL1 (latest flowering) and SIZ1 (earliest flowering)
<ul>
<li>Knockout of SIZ1 caused reduced drought tolerance. <em>H. belmoreana</em> has lower expression of the two SIZ1 homologues and also cannot colonise calcareous soils. On the other hand, knockout of SAL1 caused increased drought tolerance (consistent with <em>H. forsteriana</em>), but reduced salt tolerance which doesn’t match with <em>Howea</em> scenario—so less evidence for this locus driving sympatric speciation cf SIZ1.</li>
</ul></li>
<li>Robust example of genetic architecture of sympatric speciation. Evidence of pleiotropy in multiple loci, so could be a combination of pleiotropy and linkage disequilibrium.</li>
</ul>
<p>How did knockout (removal) give these results? Counterintuitive?</p>
</section>
<section id="dean-et-al-2019" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="dean-et-al-2019"><span class="header-section-number">3.2</span> Dean et al 2019</h2>
<ul>
<li>An assumption when considering sympatric species pairs is that they originated from a recent, genetically homogenous ancestor, but most evidence invokes introgression/admixture upon secondary contact as responsible for speciation. The authors compare two ecotypes of three-spined stickleback (no, multiple species-pairs?) using morphological data, targeted genotyping and genome-wide SNP data.
<ul>
<li>The two ecotypes show extensive hybridisation in sympatry/parapary.</li>
<li>There are rare cases of complete reproductive isolation in sticklebacks, but these cases involve ancient allopatric origins.</li>
</ul></li>
<li>Importance of genetic admixture upon secondary contact widely known in plants since 1970s, but has also started gaining popularity in animals.
<ul>
<li>Classic examples like Darwin’s finches, Lake Victoria cichlids and <em>Anopheles</em> species complex all either involved genetic admixture events or are cryptic cases of homoploid hybrid speciation.</li>
</ul></li>
<li>Overwhelming evidence for ancient allopatry, and against “true” sympatric ecological speciation. (And authors found the same.)</li>
<li>They find that the two ecotypes are strongly reproductively isolated, but also a complex colonisation history suggesting admixture upon secondary contact in addition to disruptive ecological selection (i.e., older allopatric origin).</li>
<li>Three hypotheses (none consider sympatric speciation option? wait, the first classic one probably is)</li>
<li>Compelling morphological and genetic evidence of unusually strong reproductive isolation between lagoon resident and anadromous stickleback, in spite of (low) gene flow. Classic purely-ecological speciation model (relative sea level change) unlikely; at least two colonisation and therefore admixture upon secondary contact (alongside strong selection for maintenance of differentiation, i.e.&nbsp;partial ecological basis)</li>
<li>Reproductive isolation in this island species-pair example is considerably stronger than most other stickleback examples, so this could potentially be a very useful system to work further on these questions.</li>
<li>Also evidence against “linear, bifurcating process” model.</li>
</ul>
<p>Species pairs? Or just ecotypes? Confusing.</p>
</section>
</section>
<section id="rundle-nosil-2005" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Rundle &amp; Nosil 2005</h1>
<ul>
<li>Renewed interest in microevolutionary processes driving macroevolutionary speciation was linked to reclassficiation of speciation models from geographic scheme (sympatry vs allopatry) to focus on reproductive isolation.</li>
<li>Barriers as a result of ecologically driven divergent selection</li>
<li>Ecological speciation different from other models that focus on chance events (polyploidisation, hybridisation, genetic drift, founder events, population bottlenecks), or on non-divergent and/or non-ecological selection (runaway selection, sexual conflict)</li>
</ul>
<section id="ecological-causes-of-divergent-selection" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="ecological-causes-of-divergent-selection"><span class="header-section-number">4.1</span> Ecological causes of divergent selection</h2>
<p>Environmental differences, sexual selection, ecological interactions</p>
</section>
</section>
<section id="discussion-pre-zygotic-isolation" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Discussion: Pre-zygotic isolation</h1>
<section id="zhang-et-al-2024" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="zhang-et-al-2024"><span class="header-section-number">5.1</span> Zhang et al 2024</h2>
<ul>
<li>Role of ecology and geography have not been explored much at later stages of speciation, because traditionally focus on role of ecology in <em>initiating</em> speciation and on role of geography (reinforcement) on species pairs lacking divergent characteristics.</li>
<li>How divergent ecology interacts with geography to affect RI in later stages of speciation remains unclear</li>
<li>Reinforcement: increased favour towards prezygotic isolation in sympatry, peripatry or parapatry, due to fitness cost to hybridisation</li>
<li>Increased habitat isolation can also arise from direct selection against immigrants (high fidelity to native environment favoured)</li>
<li>Three gall wasps in genus <em>Belonocnema</em>, specific to <em>Quercus</em> hosts.</li>
<li>Existing knowledge: two sympatric wasp species are in later stages of speciation</li>
<li>Test: in allopatry, does host plant divergence yield higher RI? In host plant divergence, does sympatry yield higher RI than allopatry (due to selection for reduced migration and reduced hybridisation)?</li>
<li>Measured “degree of sexual isolation in the absence of host plant”</li>
<li>Natural temporal difference in the wasp species, so they stored some galls at 4 dec C for one week which delayed the emergence. Adults were used in experiments 1-2 days after emergence.</li>
<li>Measured host plant selection using a controlled two-choice experiment, using females instead of males (female host choice determines environment for next generation, and males display lower host fidelity).
<ul>
<li>Female aspirated into the container and observed for 30 min at 2-min intervals (15 obs), with the assumption that time spent on host plant is correlated with oviposition decision.</li>
<li>Calculated proportion of time spent on natal plant (<span class="math inline">\([0, 1]\)</span>), compared using GLMM.</li>
<li>Also calculated pairwise estimates of host plant preference difference: absolute difference between preference values calculated above (also <span class="math inline">\([0, 1]\)</span>).</li>
</ul></li>
<li>Measured sexual isolation by observing courtship and mating interactions between males and females, using controlled no-choice preference trials.
<ul>
<li>Male and female (same or different species) aspirated into container, observed for same time, and recorded three interactions: male wing buzzing/fanning, male mounting female, copulation.</li>
<li>Calculated probability of mating between conspecific and heterospecific pairs, and compared using GLMM.</li>
<li>Comparison between species pair type and individual pair cannot be done using GLMM, so they also calculated strength of sexual isolation between species pairs using a formula.</li>
</ul></li>
<li>Evidence that host use (ecological) interacts with geography in driving divergent selection in <em>Belonocnema</em> wasps.
<ul>
<li>Sympatric wasp species with different host plant showed highest habitat isolation, while allopatric with same host plant showed lowest.</li>
<li>Proximity to two host plant options increased host preference (Petri dish vs cups)</li>
<li>Sympatric also showed highest sexual isolation, while allopatric with same host plant showed lowest.</li>
</ul></li>
<li>High isolation in allopatric species with same host plant suggest that selection is not directly driven by habitat use. Similar selection environment could have taken different phenotypic pathways (mutation-order speciation).</li>
<li>Three possible mechanisms for high isolation in sympatric species:
<ul>
<li>Templeton effect/differential fusion: in sympatry, strong isolation is required for isolated populations to persist, whereas weakly isolated populations fuse.</li>
<li>Reinforcement: natural selection against hybrids with lower fitness</li>
<li>Reproductive interference: presence of sympatric species during mating (or cost of migration) could elevate prezygotic isolation</li>
</ul></li>
<li>Also found asymmetry between habitat isolation and sexual isolation between allopatric species sharing the same host plant: lineage that emigrates more shows stronger habitat isolation (lower fitness of immigrants in alternative habitat) while lineage that receives more immigrants shows stronger sexual isolation (lower fitness of hybrids).</li>
<li>The three closely related wasp species are a good model for speciation research, due to their unique ecology and geography. Their framework is also useful for exploring completion of speciation, a knowledge gap.</li>
</ul>
</section>
<section id="schield-et-al-2024" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="schield-et-al-2024"><span class="header-section-number">5.2</span> Schield et al 2024</h2>
<ul>
<li>How does the genetic basis of sexual selection result in barrier to gene flow in secondary contact?</li>
<li>Most studies explore links between either sexual traits and reproductive isolation, or genomic divergence and genetic association with sexual traits. Not the full flow from genomic divergence to reproductive isolation via barrier to gene flow.</li>
<li>For species like Barn Swallow which are early in speciation process, sexual selection may be driving RI at small number of barrier loci involved in sexual traits.
<ul>
<li>Genetic coupling of barrier loci (linkage disequilibrium) facilitates speciation, via both direct and indirect selection. With increase in coupled barrier loci, barrier to gene flow is stengthened.</li>
<li>Coupling is likely to arise in allopatry, and can promote RI upon further secondary contact.</li>
</ul></li>
<li>Barn Swallow populations all diverged recently from the common ancestor about 11 kya, and there is strong evidence for divergent sexual selection among these populations.
<ul>
<li>Not only does the species have multiple populations (subspecies), but there are also multiple hybrid zones, with varying divergence in sexual traits. These hybrid zones allow tests of whether sexual selection results in RI upon secondary contact.</li>
<li>Further, two of these hybrid zones also have divergences in nonbreeding migratory route, which is a potential coupled trait (low survival of hybrids taking nonadaptive migratory routes; i.e., hybrid fitness).</li>
<li>Explicit tests of effects of trait loci on gene flow are necessary because degree of concordance between allopatric evolutionary processes and hybrid fitness is not clear a priori.</li>
</ul></li>
<li>Do sexual trait loci experience divergent sexual selection in allopatry and promote RI upon secondary contact? Is there LD among barrier loci (sexual traits, migratory route) in secondary contact, mainting RI via effects on hybrid fitness?</li>
<li>Found that 6 ssp. cluster in 3 groups. Individuals from hybrid zones show intermediate/similar to parent plumage trait values (breast feather colour, tail streamer length) and admixture proportions.</li>
<li>Prevalence of back-crossed, late generation hybrids, and demographic models, suggest that hybrid zones formed upon secondary contact following allopatry.</li>
<li><em>rustica</em>-<em>gutturalis</em> hybrid zone estimated to have formed 2000 generations ago, while the <em>tytleri</em> ones were more recent (&lt; 1000 generations).</li>
<li>Trait variation is controlled by few loci and not several genes: both traits have high proportion of phenotypic variation explained (PVE) by SNP genotypes, and high proportion of genetic effects (PGE) by small number of SNPs.</li>
<li>Ventral colour SNPs on chromosome 1A and Z (genes affecting melanogenesis and feathering phenotype), whereas tail streamer length SNPs on chromosome 2 (enzymes with regulatory functions potentially affecting transcription, intracellular transport, growth factor stability).
<ul>
<li>Concentration of ventral colour associations on Z chromosome (sex chromosome) supports that sex linkage is favourable for traits involved in mate choice and sexual isolation.</li>
</ul></li>
<li>Across allopatric populations, sexual trait loci show more genetic differentiation (FST) than do genome-wide comparisons.
<ul>
<li>This could just be due to “background selection” in regions with reduced recombination. However, regions with extremely low recombination rates like centromeres also show higher FST in trait loci—high differentiation in trait loci is due to divergent positive selection rather than background selection alone. Elevated allelic differentitation and extended haplotype homozygosity, also suggest this.</li>
<li>Greater fitness effects of ventral colour than tail streamer length.</li>
</ul></li>
<li>Geographic and genomic cline analysis, comparing trait loci–background loci differences across space along the three hybrid zone transects. -Found steeper transition between parental ancestries in trait loci cf background loci.
<ul>
<li>Steepness itself could arise due to low population density in hybrid zones, but this would show similar clines in trait loci as in other highly differentiated loci (not the case)</li>
<li>Stronger selection and barrier to gene flow between <em>rustica</em> and other ssp.</li>
<li>Cline patterns are weakest in <em>tytleri</em>-<em>gutturalis</em> hybrid zone, possibly linked to absence of migratory divide between the two populations (though remains to be tested).</li>
</ul></li>
<li>LD arises from hybridisation, but extent of erosion of LD by recombination over generations can be impacted by selection-driven genetic coupling.
<ul>
<li>They compared interchromosomal LD between trait loci and background loci; if the LD was formed by admixture but not maintained by selection, would find similar levels between the two. But they found much higher LD levels in trait loci.</li>
<li>Increase in LD shows consistent differences between hybrid zones, with <em>rustic</em> hybrids showing higher.</li>
<li>Also no differences in LD between parental populations, therefore admixture important followed by selection.</li>
<li>Starting levels of admixture LD higher than current levels, suggesting that selection worked by slowing erosion of LD, rather enhancing LD beyond starting levels.</li>
</ul></li>
<li>Primary/major role of divergent sexual selection in genomic divergence and RI.</li>
</ul>
</section>
</section>
<section id="discussion-reinforcement" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Discussion: Reinforcement</h1>
<section id="bemmels-et-al-2021" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="bemmels-et-al-2021"><span class="header-section-number">6.1</span> Bemmels et al 2021</h2>
<ul>
<li>Dynamics of allopatry initiating speciation, widely accepted. But how sympatry (sec.&nbsp;contact) can complete speciation is contested.</li>
<li>RI before secondary contact: either fully complete, or costs to interactions between partially RI species (maladaptive hybridisation) may drive further divergence (reinforcement).</li>
<li>Problem with complete RI prior to sympatry: takes very long time to evolve, and widepsread occurrence of hybrid zones suggests frequent contact with incomplete RI.</li>
<li>Alternative is that incipient species interactions change (indirect)—any process that results in divergent character displacement, e.g., divergent agonistic selection to avoid aggressive interactions (though this could result in the opposite, convergence, if interspecific aggression helps defend against true competitors).</li>
<li>Another option is direct selection for increased RI (direct): reinforcement, i.e., selection to avoid maladaptive hybridisation. Very difficult to demonstrate its existence, so few examples. One of the hallmark indicators of reinforcement is reproductive character displacement (increased divergence in reproductive traits in sympatry than in allopatry), but this can be generated by other processes too.</li>
<li>Difficult to determine whether sympatry plays a role in completion of speciation (geographic context difficult to estimate).</li>
<li>Coalescent demographic modelling, whole-genome datasets. Sympatric E NA <em>Empidonax</em> species populations (broad sympatry) show increased RI post secondary contact—unrelated to reproductive character displacement—while western contact zone (narrow sympatry) has lower RI (higher introgression).</li>
<li>Methods
<ul>
<li>Whole genome data –&gt; patterns of admixture (is RI higher in broad sympatry than in narrow sympatry?), and patterns of ecological/reproductive character displacement in sympatry. Coalescent modelling –&gt; did geographic contact precede and drive increase in RI?</li>
<li>Blood samples –&gt; whole-genome sequencing and genotyping-by-sequencing –&gt; alignment with Willow reference genome, generation of de no Alder genome (linked-read sequencing).</li>
<li>Genomic differentiation: weighted Hudson’s F_ST_, proportion of genetic ancestry.</li>
<li>Geographic contact: GBIF, WorldClim, Maxent SDMs for present climate and LGM. (Potential for range overlap.)</li>
<li>Character displacement: morphological measurements of caught birds. Temporal displacement from spring arrival time from eBird.</li>
</ul></li>
<li>Genomes are well differentiated, multiple F_ST_ peaks on most chromosomes, greater in east than west. Similarly, admixture lower in east than west. RI in east not absolute but nearly complete and significantly higher than in west.</li>
<li>Longer contact in east than in west, south of glaciation. Willow occurred in both but Alder (currently more N breeder) only in east. (However, range overlap in LGM not really in “east”, more central [Fig 3].)</li>
<li>No temporal character displacement. No displacement in bill/body morphology or crown feather reflectance. Only difference in crown colour—subtle, but strongest in UV, but birds unlikely to perceive this difference.</li>
<li>Support for continuous gene flow even in present day in west, but ceased long after secondary contact in east. However, model support relatively weaker in west (additional complexity).</li>
<li>Elevated RI could have been due to ecological divergence to avoid resource competition, agonistic selection, or reinforcement—but causal traits elusive.</li>
<li>Multistage model of speciation: dynamics of initiation and completion of RI have fundamentally different dynamics and drivers.</li>
<li>Potential alternative reproductive traits that could be under divergent selection: male song, and female preference for male song. Is divergence of male song stronger in sympatry? Is female preference for conspecific song higher in sympatry (reinforcing selection) as in <em>Ficedule</em>? Or even gametic level (sperm-related). Also need to explore costs to hybridisation like reduced fitness.</li>
<li>Their approach is one effective way to study what otherwise remains very difficult, even in non-model systems and species that no longer hybridise.</li>
<li>Broad sympatry vs narrow sympatry —&gt; sympatry vs allopatry (not sure if this logical jump is appropriate)</li>
</ul>
</section>
<section id="levitan-hao-2025" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="levitan-hao-2025"><span class="header-section-number">6.2</span> Levitan &amp; Hao 2025</h2>
<p>About postzygotic (gametic) isolation</p>
<ul>
<li>How do initially rare reproductive genotypes proliferate over time if they by definition result in reduced mate compatibility?</li>
<li>External fertilisation –&gt; ease of study, wide variation in intraspecific gametic compatibility.</li>
<li>Possibility: sexual conflict over fertilisation rate selects for eggs with reduced compatibility, producing population with more than one matched compatibility group. But although this generates variation, not sufficient to explain very low compatibility preventing/limiting hybridisation.</li>
<li>Two urchin species can produce F1 embryos; <em>Sd</em> eggs more susceptible than <em>Mf</em>; former can metamorphose into juveniles but magnitudes lower than conspecifics (no introgression between the two species).</li>
<li>Tested for conspecific sperm precedence (CSP) using no-choice and competitive choice assays. Is variation in hybrid fertilisation related to male and female gamete proteins? Are derived protein variants less likely to result in hybrid fertilisation than ancestral? Do allele frequencies of sperm and egg proteins in sympatry and allopatry support presence of RCD?</li>
<li>No-choice fertilisation success between conspecific and heterospecific males and females reciprocally. Choice experiments using <em>Sd</em> eggs, because it has higher and more variable levels of hybridisation than <em>Mf</em>.</li>
<li>Experiment 1: no-choice + 5 choice (varying heterospecific sperm ratios) crosses. Experiment 2: no-choice + 0.9:0.1 (<em>Mf</em>:<em>Sd</em>) choice crosses (ability of sperm to fertilise eggs at greatest numerical superiority; more replicates).</li>
<li>In choice crosses, embryos developed for 3 days then genetic analysis for paternity. Survivorship compared among crosses to test for bias between early postzygotic survivorship (prior to genotyping larvae) and gametic incompatibility.</li>
<li><em>Sd</em> more susceptible to hybrid fertilisation than <em>Mf</em>, and required lower concentration of heterspecific sperm. But presence of conspecific sperm reduced hybridisation—47% no hybridisation even at most extreme ratios favouring hybridisation. Concentration no longer important when both sperm present; instead, relative proportion important.</li>
<li>Above were overall/average trends, but individual results varied from absolute to nonexistent CSP. This variation was explained by the sperm genotype: presence of shared (ancestral) GG allele increased hybridisation. Higher hybridisation in presence of shared protein was seen also in egg protein, but this was not statistically significant.</li>
<li>Shared allele more frequent in allopatry than sympatry, but only in eggs and not in sperm bindin (greater cost of egg wastage than sperm wastage).</li>
<li>Interspecific variation in sperm recognition proteins not enough to fully prevent hybrid fertilisation if given enough time and without competing sperm with higher affinity (conspecifics).</li>
<li>Likely that reinforcement selection and RCD manifested more in <em>Mf</em>.</li>
<li>Sexual conflict, or generally sexual selection, generates intraspecific variation which then gets acted upon by reinforcement selection.</li>
<li>Did not like the way this paper was written:
<ul>
<li>Background on the proteins: important info but could have had lay summaries too.</li>
<li>Overll just too technical!</li>
<li>Results too long, longer than discussion, and contains things that should be in methods (like models and tests)!</li>
<li>Paper should overall have been more inferential, rather than technical reporting.</li>
</ul></li>
</ul>
</section>
</section>
<section id="discussion-genetic-basis-of-speciation-i" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Discussion: Genetic basis of speciation I</h1>
<section id="turbek-et-al-2021" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="turbek-et-al-2021"><span class="header-section-number">7.1</span> Turbek et al 2021</h2>
<ul>
<li>The role of assortative mating (premating isolation), phenotypic traits of species recognition and their genomic basis and origin.</li>
<li>No ecological barriers yet (and low genomic divergence), but behavioural isolation.</li>
<li>Viable hybrids are readily formed, indicating lack of genetic incompatibility.</li>
<li>Post-mating isolation requires longer time in general than pre-mating isolation mechanisms.</li>
<li>Genomic divergence using shotgun short-read whole-genome sequencing</li>
<li>PCR amplification for Sanger sequencing of short genomic region having high divergence</li>
<li>Choice experiments using song playback and artificial mounts of conspecific and heterospecific males. Used sympatric and ecologically similar heterospecific control.</li>
<li>Genomic divergence peaks contain plumage colouration genes.</li>
<li><em>Sh</em> segregating within <em>Si</em> at high differentiation sites, suggesting either incomplete lineage sorting or past hybridisation.</li>
<li>Female colouration largely overlaps between the two species even in avian visual spectrum.</li>
<li>Paired males and females showed the same F_ST peaks in whole-genome and double-digest restriction site–associated DNA (ddRAD) sequence data, suggesting lack of hybrid pair formation.</li>
<li>Although there was high extra-pair mating, candidate fathers and social fathers were of the same species. Social and extra-pair mating maintain assortative mating.</li>
<li>Genomic differentiation from phylogenies for the whole capuchino radiation suggested recent speciation because there was no species-level monophyly. However, <em>Si</em> and other restricted range species formed clades. On the other hand, in phylogenies derived from only the divergence peak regions, it did not form a clade. Although many species shared variants with it at individual peaks, none showed the specific combination (both divergent peaks) of <em>Si</em>. This indicates that <em>Si</em> phenotype likely arose from standing genetic variation.</li>
<li>Assortative mating through genetically inherited (plumage, song?) and perhaps culturally inherited (song?) traits, at least early in speciation until subsequent reproductive barriers arise.</li>
<li>Important divergence peaks are on Z chromosome (large-Z effect).</li>
</ul>
</section>
<section id="raffini-et-al-2025" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="raffini-et-al-2025"><span class="header-section-number">7.2</span> Raffini et al 2025</h2>
<ul>
<li>Inferences about accumulation of RI depend on comparing contemporary populations that are at different stages on a “speciation continuum”. Cannot assume a monotic progression from weak to strong isolation.</li>
<li>Important to distinguish between different stages of RI from differences among taxa</li>
<li>Compare two ecotypes in same marine snail species, showing similar levels of phenotypic divergence (two ecotypes in both regions), but different levels of genomic divergence (and gene flow/isolation). Parallel speciation.</li>
<li>The two ecotypes occur close together on many shores and also form contact zones. Spanish (older) and Swedish (younger, after last glaciation).</li>
<li>Differences between ecotypes have been suggested by several studies, but detailed data combining genome-wide differentiation and spatial coverage of contact zones has been lacking.</li>
<li>Dense transects across contact zones, low-coverage whole-genome sequences, shell features, behavioural traits.</li>
<li>Unique: Groups identified/assigned genetically, not based on phenotypic traits.</li>
<li>Patterns of genomic divergence explored using PCA and DAPC</li>
<li>Patterns of chromosomal inversion, known to contribute to the differentiation of these two ecotypes, were explored using unthinned datasets. First used Manhattan plots of per-locus F_ST between genetic clusters, with randomisation 1000 times. Genomic inversions differing in frequency between clusters would show higher differentiation than non-inverted regions. Second, used PCAs of linkage map position.</li>
<li>Some phenotypic traits using standardised image analysis. Cluster analysis and PCA.</li>
<li>Both regions showed genomic divergence (PCA of SNPs) between ecotypes, but patterns of genetic differentiation between ecotypes were different in the two regions (Sweden more gradual cline, Spain two genetically distinct clusters with no intermediate genotypes and with wave ecotype spatially clustered in low parts of shores). Divergence between Spanish ecotypes larger than divergence among all Swedish.</li>
<li>Role of chromosomal inversions: Found multiple chromosomal inversions at same sites across countries. Although inversion patterns were congruent between the countries, more pronounced in Spain (higher blocks in Manhattan plots, stronger cluster in PCA).</li>
<li>Strong phenotypic and behavioural (Crab more wary) divergence between two ecotypes, but strength lower in Sweden where Crab traits more often also seen in Wave than is the case in Spain.</li>
<li>Phenotypic variation showed even more gradual cline in Spain than genomic. Sweden showed higher phenotypic divergence between ecotypes than Spain, though both in same direction.</li>
<li>While two regions share many similarities, they also show several important differences suggesting that they are at different stages of the speciation continuum: Spanish closer to completion of speciation.</li>
<li>Sex determination was on different loci in the two countries, intraspecific variability in genetic determination of sex.</li>
<li>Size might be a magic trait, especially in Spain where habitat is more patchy and less gradual.</li>
<li>Completion of speciation involves complex ecological and/or evolutionary dynamics.</li>
</ul>
<p>Is there evidence that this divergence happened independently in the two regions? Relatively young species, and young Swedish population. Found low gene flow between overlapping ecotypes in low shore, but I wonder how that looks relative to gene flow between Crab ecotypes at different points on the shore?</p>
</section>
</section>
<section id="discussion-genetic-basis-of-speciation-ii-gene-regulation" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Discussion: Genetic basis of speciation II: Gene regulation</h1>
<section id="chen-et-al-2024" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="chen-et-al-2024"><span class="header-section-number">8.1</span> Chen et al 2024</h2>
<ul>
<li>Residual sequences of ancient retroviruses influence the adaptive diversification of species by regulating host gene expression.</li>
<li>ERVs are typically non-coding, having lost most their function over time. However, they still introduce novel genetic material and biological function in hosts. Therefore, acquisition of ERVs may have been adaptive and contributed to the evolution of novel traits.</li>
<li>In birds, which have generally smaller genomes than near outgroups, estimates of LTR elements of ERV genome have been low. Biological function of solo-LTR poorly known.</li>
<li>Found higher frequencies of solo-LTR formation (higher efficiency in purging EVE from genome), continuous accumulation of solo-LTR through speciation of Passeriformes (data on transcription expression and population resequencing in zebra finches), 20 genes in zebra finches with lineage-specific solo-LTR differentially expressed cf orthologs in chicken, dual-luciferase reporter assays confirm cis-regulatory activity of solo-LTRs upstream of ITGA2 gene.</li>
<li>Solo-LTR formation as host defence mechanism or result of high recombination activity among LTRs. Additionally, many solo-LTRs carry regulatory elements and have cis-regulatory function.</li>
</ul>
</section>
<section id="yusuf-et-al-2025" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="yusuf-et-al-2025"><span class="header-section-number">8.2</span> Yusuf et al 2025</h2>
<p>(Much more dense and difficult to follow. Simple, interpretive language, guys!)</p>
<ul>
<li>Cis-regulation known to influence morphological and physiological RI, but not much known about mate recognition behaviour, specifically about genetic mode of action driving signals and preferences.</li>
<li>Allele-specific expression differences (differences in expression of gene copies from maternal vs paternal genomes) across conditions. F1 hybrid female neural samples grouped by parental allele, indicating strong cis-regulatory divergence in neural expression patterns between species irrespective of acoustic environment.</li>
<li>Found extensive, genome-wide CRE divergence related to response to male acoustic sexual signals. (Broad distribution of diverged CR genes across chromosomes, differences in function of genes having context-dependent divergent expression, genomic signatures of divergent selection in allopatry.)</li>
<li>Notably more CRE divergence in sympatry than allopatry, suggesting that neurogenetic pathways of mate recognition are social context-dependent.</li>
</ul>
</section>
</section>
<section id="discussion-genome-architecture" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Discussion: Genome architecture</h1>
<section id="meyer-et-al-2024" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="meyer-et-al-2024"><span class="header-section-number">9.1</span> Meyer et al 2024</h2>
<ul>
<li>Large structural variants (SVs) like inversions both establish and maintain divergent haplotypes (and reducing inter-haplotype recombination).</li>
<li>Although popular in studies of differentiation (and ecotype formation), still don’t know what conditions give rise to inversions, how they are maintained over the long term, and how they contribute to RI and speciation.</li>
<li>LD can form between multiple distant functional sites, but also between co-adapted gene complexes and deleterious mutations (by chance). So dynamics of inversion can be driven by multiple simultaneous processes.</li>
<li>Allelic content of inversion can also change over time: new mutations, or gene flux between inverted haplotypes. This can influence the ferquency of an inversion and ultimately the evolutionary trajectory (and contribution to RI).</li>
<li>N-S lineages in Northeast Atlantic and marine-lagoon lineages in Mediterranean. Strong genetic differences in the same part of the genome, meaning genomic architecture probably relies on shared structural variation.</li>
<li>Whole-genome resequencing of samples; high-coverage linked-read sequencing for reference genome. PCA on SNPs for population structure. Genetic differentiation (F<sub>ST</sub>), nucleotide diversity (<span class="math inline">\(\pi\)</span>), absolute genetic divergence (d<sub>xy</sub>) to characterise genomic landscape of divergence. Ancestral recombination graphs (ARGs) to describe variation in coalescence times across genome using time to the most recent common ancestor (TMRCA); local ancestry for each haplotype within inversion regions.</li>
<li>Pronounced population structure even in adjacent sampling locations; tight LD. Clustering by geography is only for LD regions, overall clustering seems to be based on chromosomal inversions (three inversion genotypes), suggesting two polymorphic inversions containing large numbers of sites in strong LD. Similarly, genome-wide differentiation is relatively weak, but F<sub>ST</sub> and d<sub>xy</sub> values were high in two chromosomes (2, 12).</li>
<li>Sufficient time has elapsed since appearance of inverted haplotypes to have mutation-drift equilibrium.</li>
<li>Origin of inversions: inversions appeared within the lineage with no introgression or gene flow upon secondary contact—instead, incomplete lineage sorting (inversion in B12 is near chromosome extremity), remained polymorphic for long period of time.</li>
<li>B12 is type 1 (divergent) but B2 involves interaction of Type 1 and Type 2 (within-population variation). B2 inversion is private to Mediterranean lagoons: intermediate haplotype frequencies, heterozygote advantage; because it is longer, more likely to have accumulated deleterious mutations, so polymorphism maintained by pseudo-overdominance.</li>
<li>B12 might facilitate speciation over the long term, but uncertain whether B2 will undergo differential fixation between habitats or universal fixation of one haplotype.</li>
</ul>
</section>
<section id="huang-et-al-2025" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="huang-et-al-2025"><span class="header-section-number">9.2</span> Huang et al 2025</h2>
<ul>
<li>Because inversions tightly link locally adapted alleles, adaptive response is poor when the population encounters a new environment.</li>
<li>Dunes have low nitrogen levels, low vegetation cover, than surrounding sand sheet. Dune ecotype has larger/heavier seeds.</li>
<li>Habitat data + genotyping-by-sequencing data –&gt; genotype-environment association (GEA) analyses to test for associations between inversions and shared selective pressures. QTL mapping to identify seed size difference regions. Whole-genome sequences (existing), population genomics to identify selective sweeps shared by two dune ecotypes (do inversions contribute disproportionately to parallel genetic evolution).</li>
<li>Independent origins of the two dune ecotypes, because SNP PCA yielded geographical clusters of both dune and non-dune. This is despite habitat variables (soil and vegetation) separating between dune and non-dune (PC1).</li>
<li>Crosses on seed size had F<sub>1</sub> with intermediate seed size. In F<sub>2</sub>, many genomic regions were overrepresented in large-seeded plants, meaning seed size is highly polygenic.</li>
<li>QTLs associated with seed size belonged to inversions more likely than not, and more QTL windows were shared than expected by chance in pairs within and between locations.</li>
</ul>
</section>
</section>
<section id="discussion-introgression" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Discussion: Introgression</h1>
<section id="jensen-et-al-2024" class="level2" data-number="10.1">
<h2 data-number="10.1" class="anchored" data-anchor-id="jensen-et-al-2024"><span class="header-section-number">10.1</span> Jensen et al 2024</h2>
<ul>
<li>Due to Haldane’s rule, gene flow between divergent lineages more likely through homogametic hybrids than heterogametic sex—leading to phylogenetic discordance between autosomes and homogametic sex loci.</li>
<li>Y chromosome introgression therefore rare, and the few cases reported have mostly been in relatively recently diverged lineages. Related to positive selection and meiotic drive.</li>
<li>In guenons, which hybridise a lot in the past and present, males disperse after maturation, giving opportunities for introgression across species boundaries. Maternal introgression cases already known in these organisms.</li>
<li>WGS two guenon species (<em>Cercopithecus denti</em>, <em>C. wolfi</em>) to add to recent dataset, and resequenced two others.</li>
<li>Why compared against rhesus macaque? Why not another species from the same continent, near relative?</li>
<li>Saw mitonuclear discordance, but Y tree was very similar to species tree, in line with Haldane’s rule—except for <em>C. denti</em> which in the Y tree was closest to <em>C. mitis</em> and not <em>C. wolfi</em>.</li>
<li>To distinguish between introgression and ILS, compared pairwise nucleotide differences (d<sub>xy</sub>), and found that nucleotide divergence was 3.6 times greater on autosomes than Y (suggesting introgression from <em>mitis</em> into <em>denti</em>).</li>
<li>Autosomal gene flow tested using D-statistic, and found overall low levels of allele sharing between <em>mitis</em> and <em>denti</em>, suggesting that the <em>mitis</em>-like Y chromosome in <em>denti</em> was initially introduced at a low frequency.</li>
<li>Then compared neutral (drift) and non-neutral processes using simulations (multi-species-coalescence-with-introgression, MSci). Low initial frequency but selective force allowed fixation.</li>
<li>Looked for possible targets of selection in Y-linked genes with amino acid substitutions in the different lineages. Found potential in genes linked to sperm competition.</li>
<li>Tried to compare adaptive introgression with meiotic drive, but didn’t get conclusive results that suggest one over the other.</li>
</ul>
</section>
<section id="jorquera-et-al-2025" class="level2" data-number="10.2">
<h2 data-number="10.2" class="anchored" data-anchor-id="jorquera-et-al-2025"><span class="header-section-number">10.2</span> Jorquera et al 2025</h2>
<ul>
<li>Three skua species in S Hemisphere with rather distinct breeding ranges; generalist scavengers and predators. Failure to resolve phylogenetic status/taxonomic relationships of Brown Skua subspecies (ILS, introgression, methodology limitations).</li>
<li>Clarified evolutionary history of SH skuas (reconstructing demographic history), revealing timing of diversification and impact of past climate oscillations on effective population size. Then, used SDMs to predict range changes in areas of introgression.</li>
<li>Used high-coverage (15x) WGS from 111 skuas in 21 locations, covering entire SH spatial extent of their ranges.</li>
<li>Genetic structure and differentiation (using SNPs) suggest three clusters, aligning with current species groups. But also show several admixed populations. Phylogeny and population structure based on nuclear SNPs show 7 phylogeographic groups.</li>
<li>Reconstrcuted phylogeny using TreeMix, which suggests gene flow events in the past (not just branching tree).</li>
<li>D-statistics and f-branch test to distinguish between ILS and introgression.</li>
<li>Genomic signatures of adaptation using RAiSD and XP-nSL: detect genome-wide selective sweeps that produce regions of reduced genetic diversity near an adaptive mutation. Generally, chromosomes 9 and 18, with broad cellular functions. Strongest differential selection, nearing fixation, associated with growth, reproductive processes, and pigmentation.</li>
<li>Specifically, might be related to thermogenic adaptations. Lipid metabolism in Brown Skua, fatty acid degradation and metabolic pathways in SP Skua, glycerolipid metabolism in Chilean. Traits: ability to endure extreme temperatures, prolonged periods without food, various environmental pressures.</li>
<li>SDMs predict changes for all species, and this will change overlaps and therefore introgression. Balance between disruptive selection and homogenisation via gene flow will determine future genetic differentiation, and future of the species groups.</li>
<li>Very weirdly written paper, hotchpotch (flow is poor). Even discussion ends abruptly.</li>
</ul>
</section>
</section>
<section id="discussion-old-variation" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Discussion: Old variation</h1>
<p>(not detailed reads this time)</p>
<section id="meier-et-al-2023" class="level2" data-number="11.1">
<h2 data-number="11.1" class="anchored" data-anchor-id="meier-et-al-2023"><span class="header-section-number">11.1</span> Meier et al 2023</h2>
<ul>
<li>African lake cichlids: largest radiation was in a very young lake, and all ecological guilds evolved in situ through lineage fusion (admixture) and fission (speciation), from several ancestral populations that met.</li>
<li>LVRS cichlid fishes had a hybrid origin: 2-3 older lineages that had split from common ancestor through geographical isolation</li>
<li>Should Lake Victoria clades all be considered monophyletic, instead of having same genera across lakes?</li>
<li>Entire LVRS had hybrid origin between Congolese and Upper Nile lineages. Victoria Radiation formed from hybridisation when lake refilled. Later, hybridisation within VR gave rise to ecological groups and hybrid species with novel trait combinations.</li>
<li>Syngaemon dynamics similar to metapopulation dynamics.</li>
<li>Natural hybrids should not be dismissed as detrimental or evolutionary dead-ends.</li>
</ul>
</section>
<section id="lutgen-et-al-2025" class="level2" data-number="11.2">
<h2 data-number="11.2" class="anchored" data-anchor-id="lutgen-et-al-2025"><span class="header-section-number">11.2</span> Lutgen et al 2025</h2>
<ul>
<li>Wheatears show high incidence of convergence plumage colour evolution and frequent hybridisation. Same phenotypes reappear multiple times across the genus.</li>
<li>In <em>hispanica</em> complex, throat colour is polymorphic in 3 of 4 species, while mantle/neck colour is recombined by hybridisation.</li>
<li>Variation in throat and mantle colouration are associated with agouti signalling protein (ASIP) locus</li>
<li>Combination of new and old/introgressed variation at a single gene. This (complex recombined phenotypes underpinned by monogenic architecture) is rare.</li>
<li>Recombined phenotypes (due to introgression) in extant hybrid zones may be a source for future phenotypic evolution</li>
<li>Even if phenotypes at different body parts are monogenic, they can be decoupled</li>
<li>Transspecific throat colour polymorphism is associated with alternative foraging niches (different N isotope signatures)</li>
<li>Abundance of genotypic and phenotypic admixture: pervasive hybridisation in the complex</li>
<li>White-throated individuals prey on insects from narrower and lower trophic level than black-throated. Thus, throat polymorphism is related to trophic niche of prey and thereby foraging strategy.</li>
<li>White mantle evolved adaptively (selection involved)</li>
<li>Story: originally, all members of <em>hispanica</em> complex had black throat and mantle, then white mantle evolved in <em>melanoleuca</em> through upstream mutations fixed by positive selection. Later, new variation introduced white throat in <em>melanoleuca</em>, which then introgressed into <em>hispanica</em> (along with white mantle, LD) and maintained as balanced polymorphism. In <em>pleschanka</em>, introgression introduced white throat but not white mantle (selected against).</li>
<li>Overall, introgression shuffled plumage colouration and led to convergent evolution.</li>
</ul>
</section>
</section>
<section id="discussion-microbes-and-microbiomes" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Discussion: Microbes and microbiomes</h1>
<section id="wang-et-al-2021" class="level2" data-number="12.1">
<h2 data-number="12.1" class="anchored" data-anchor-id="wang-et-al-2021"><span class="header-section-number">12.1</span> Wang et al 2021</h2>
<ul>
<li>Microbiome is a source of phenotypic and evolutionary novelty for hosts, since they can respond more rapidly to new selective pressures and therefore provide fitness benefits. But drastic changes in microbiome can also be problematic, and lead to dysbiosis.</li>
<li><em>Nasonia vitripennis</em> grown over 85 generations under subtoxic levels of atrazine herbicide(!), and monitored both microbiome composition and function, and host physiology and genetic differentiation.</li>
<li>16S rRNA gene amplicon sequencing of microbiome of two wasp populations over 85 generations, for genetic divergence.</li>
<li>qPCR for monitoring bacterial density.</li>
<li>After 24 generations, both groups split into two subpopulations then fed opposite diets.</li>
<li>Exposure mediated adaptive changes within the microbiome (efficient metabolism of atrazine) leading to resistance, lasting for more than 11 generations after removal of atrazine.</li>
<li>Untargeted metabolomic analysis by LC-MS, to understand how exactly microbiome mediated host resistance to atrazine. Found that hosts with the evolved microbiome more efficiently detoxified atrazine.</li>
<li>F<sub>ST</sub> different in two populations, suggesting gradual genetic divergence, and this was corroborated by ADMIXTURE plot which identified two clusters.</li>
<li>Immune challenge experiments, to test host immunity differences: heat-inactivated bacteria into wasp pupae, then used proportion of eclosed adults in 120 h as inverse measure of immune activation (induced response would channel energy away from metamorphosis).</li>
<li>Also induced selective pressure on host genome (altered gene expression and immune response). Both host and microbiome adapted, either independently or in synergy.</li>
<li>Microbiome transplants to test compatibility without coadaptation. Found an asymmetric host genome–microbiome incompatibility, suggesting that coadaptation was necessary for host to tolerate new microbiome which in turn might have resulted in differential expression and fixation of specific host genes involved in microbiome regulation.</li>
<li>Throughout, comparisons with germ-free groups helped attribute differences to microbiome.</li>
<li>They identified signatures of selection and adaptation in holobiont and hologenome, at the physiological, transcriptional and genomic levels.</li>
<li>Selection may act at multiple levels on any organism: on the host, on the microbiome, or on both in synergy. This means host-microbiome equilibrium is important and changes with adaptation.</li>
</ul>
</section>
<section id="baiz-et-al-2024" class="level2" data-number="12.2">
<h2 data-number="12.2" class="anchored" data-anchor-id="baiz-et-al-2024"><span class="header-section-number">12.2</span> Baiz et al 2024</h2>
<ul>
<li>What happens to diverged host microbiomes when host populations come in secondary contact?
<ul>
<li>Extension of DMI: intimate host-bactera relationships could mean deleterious combinations or incompatible interactions.</li>
<li>On the other hand, beneficial mutations can spread via adaptive introgression, and admixture can create opportunities for selection. Intermediate environment in hybrids may allow wider diversity of microbes than each parent alone.</li>
</ul></li>
<li>Syntopic sister taxa, which have diverged mostly in melanin-based and carotenoid-based pigmentation, but hybridise in contact zones.</li>
<li>Aimed to identify species-specific bacteria, quantify differences in gut microbiome diversity, and test whether admixed individuals exhibit disrupted microbiome diversity and whether microbiome variation is correlated with host admixture.</li>
<li>16S amplicon sequencing from faecal samples to quantify gut microbiome diversity (alpha and beta).</li>
<li>Extent of host admixture was assessed visually using plumage phenotype, which are known to be nearly perfectly correlated, summarised as a 0–1 proportion based on ordinal scores for 9 individual traits.</li>
<li>ASV abundance was higher in admixed individuals more similar to BWWA, which are more yellow (more carotenoids), suggesting that the ASVs may be linked to carotenoid deposition.</li>
<li>Overall, found low (but one ASV each) specific-specific divergence in host microbiome, unlike previous study on 15 other warbler species (here, year and locality explained more variation). They speculate that this is due to unique evolutionary history of <em>Vermivora</em>: relatively young and high genomic homogenisation due to hybridisation. Or, lack of habitat divergence could overpower species-specific effects.</li>
<li>GWWA had <em>Rickettsia</em> ASV, which showed evidence of additive genetic effects correlated with plumage score, with preference for more GWWA ancestry. On the other hand, BWWA had <em>Kineococcus</em> ASV, which is orange and encodes for carotenoid synthesis genes, and seemed to have high affinity for non-admixed BWWAs. Gut microbiome variation (abundance of ASVs) is associated with degree of host admixture.</li>
<li>But admixed individuals did not host both species-specific parental ASVs, and abundance of some ASVs is correlated with degree of admixture.</li>
<li>No sign of dysbiosis.</li>
</ul>
</section>
</section>
<section id="discussion-macroevolution" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Discussion: Macroevolution</h1>
<section id="heuer-et-al-2024" class="level2" data-number="13.1">
<h2 data-number="13.1" class="anchored" data-anchor-id="heuer-et-al-2024"><span class="header-section-number">13.1</span> Heuer et al 2024</h2>
<ul>
<li>Carnivores have extensive colour polymorphism. This is thought to confer several evolutionary advantages, especially in colonising new niches.</li>
<li>Tested predictions that speciation rates are higher in polymorphic lineages, and their divergence times are more recent, and that within closely related groups the polymorphic lineages are ancestral and monomorphic are dervied.</li>
<li>They also checked whether greater speciation rates are associated with niche breadth (no. of occupied habitats and range size).</li>
<li>Used phylogenetic comparative methods using recent Carnivora phylogenetic, time-calibrated supertree.</li>
<li>Tip branch lengths compared, then Wilcoxon rank tests and Pearson’s contingency coefficients.</li>
<li>Speciation rates estimated using Diversitree package, with multiple models: binary-state speciation and extinction (BiSSE), HiSSE, QuaSSE, SecSSE.</li>
<li>Tested niche breadth using Wilcoxon and Kruskal-Wallis tests, then somehow 95% CI and t-test…? PGLMMs with everything as random effect, then R<sup>2</sup>…? All stats are so fishy.</li>
<li>Branch lengths not significantly different between two; polymorphic carnivores higher speciation rate than monomorphic; niche breadth not important.</li>
<li>Other selective forces like predator-prey coevolution could be at play. (Suggestions of pleiotropy.)</li>
<li>Suggest that polymorphism could facilitate sympatric speciation via temporal niche segregation or assortative mating.</li>
</ul>
</section>
<section id="title-et-al-2024" class="level2" data-number="13.2">
<h2 data-number="13.2" class="anchored" data-anchor-id="title-et-al-2024"><span class="header-section-number">13.2</span> Title et al 2024</h2>
<ul>
<li>Evolution of snakes within scaled reptiles is interesting, because distinct from lizards in several ways.</li>
<li>Simpsonian macroevolution is when major biodiversity expansions occur through qualitative phase shifts into new adaptive zones, and snakes potentially align with this.</li>
<li>These phase shifts can be considered as singularities: rapid changes along many axes that appear to have been instantaneous at geological time scales, but also those that are unpredictable from prior states.</li>
<li>Authors explored the role of such phenotypic shifts in the evolutionary diversification of squamate reptiles.</li>
<li>Combined their own sequenced phylogeny with GenBank, then time-calibrated the species-level dated phylogenomic tree.</li>
<li>Traits obtained from literature and museum specimens.</li>
<li>Index of net innovation, absolute difference in phenotype between focal species and (inferred) ancestral state, separate for each trait. Another index of absolute branch-specific change along branches.</li>
<li>TR statistic similar to diversification rate, ordination to identify nodes best explaining species-level phenotypic variation.</li>
<li>Snakes have undergone major phenotypic transformations, and have much higher net innovation rates than lizards. This means the origin or early stages of snakes saw massive shifts in traits for feeding, locomotion, sensing; much of the phenotypic variation explained by one singular event early on.</li>
<li>Morphological evolution mirrored by ecological trait diversification: trophic niche evolution and diet divergence.</li>
<li>Overall, snakes cover wider dietary breadth than lizards, including aquatic and semiaquatic prey. However, diets of individual snake species are narrower than those of lizard species. Snakes also feed much less than lizards on adults of insects.</li>
<li>While morphology and diet seem to have diverged in snakes from lizards, climatic niche and life history haven’t changed much.</li>
<li>Snakes show higher speciation rates than lizards (CLaDS, BAMM), but this is not linked to richness or individual traits. In latter case, potential statistical confound with the singularity at the origin of snakes and the most successful Colubriformes radiation.</li>
<li>Snakes are an ongoing radiation characterised by sustained ecological and species (adaptive) diversification, earlier on triggered by one or more singular events.</li>
<li>Importance of dramatic shifts in tempo and mode; extrapolating contemporary microevolution through geological time may not give us the right answers.</li>
</ul>
</section>
</section>
<section id="discussion-macroevolution-1" class="level1" data-number="14">
<h1 data-number="14"><span class="header-section-number">14</span> Discussion: Macroevolution</h1>
<p><strong>2025-12-02</strong></p>
<ul>
<li>Pianka interesting guy; blew off his leg at 13 with bazooka shell</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>