<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Reading notes – BIOL 7117: Speciation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6b3bef205c96c4f74d1e1b8d3d254fec.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIOL 7117: Speciation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./class-notes.html"> 
<span class="menu-text">Class notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./readings.html" aria-current="page"> 
<span class="menu-text">Reading notes</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#mallet-2013" id="toc-mallet-2013" class="nav-link active" data-scroll-target="#mallet-2013"><span class="header-section-number">1</span> Mallet 2013</a></li>
  <li><a href="#foote-2018" id="toc-foote-2018" class="nav-link" data-scroll-target="#foote-2018"><span class="header-section-number">2</span> Foote 2018</a></li>
  <li><a href="#discussion-geography-of-speciation" id="toc-discussion-geography-of-speciation" class="nav-link" data-scroll-target="#discussion-geography-of-speciation"><span class="header-section-number">3</span> Discussion: Geography of speciation</a>
  <ul class="collapse">
  <li><a href="#coathup-et-al-2024" id="toc-coathup-et-al-2024" class="nav-link" data-scroll-target="#coathup-et-al-2024"><span class="header-section-number">3.1</span> Coathup et al 2024</a></li>
  <li><a href="#dean-et-al-2019" id="toc-dean-et-al-2019" class="nav-link" data-scroll-target="#dean-et-al-2019"><span class="header-section-number">3.2</span> Dean et al 2019</a></li>
  </ul></li>
  <li><a href="#rundle-nosil-2005" id="toc-rundle-nosil-2005" class="nav-link" data-scroll-target="#rundle-nosil-2005"><span class="header-section-number">4</span> Rundle &amp; Nosil 2005</a>
  <ul class="collapse">
  <li><a href="#ecological-causes-of-divergent-selection" id="toc-ecological-causes-of-divergent-selection" class="nav-link" data-scroll-target="#ecological-causes-of-divergent-selection"><span class="header-section-number">4.1</span> Ecological causes of divergent selection</a></li>
  </ul></li>
  <li><a href="#discussion-pre-zygotic-isolation" id="toc-discussion-pre-zygotic-isolation" class="nav-link" data-scroll-target="#discussion-pre-zygotic-isolation"><span class="header-section-number">5</span> Discussion: Pre-zygotic isolation</a>
  <ul class="collapse">
  <li><a href="#zhang-et-al-2024" id="toc-zhang-et-al-2024" class="nav-link" data-scroll-target="#zhang-et-al-2024"><span class="header-section-number">5.1</span> Zhang et al 2024</a></li>
  <li><a href="#schield-et-al-2024" id="toc-schield-et-al-2024" class="nav-link" data-scroll-target="#schield-et-al-2024"><span class="header-section-number">5.2</span> Schield et al 2024</a></li>
  </ul></li>
  <li><a href="#discussion-reinforcement" id="toc-discussion-reinforcement" class="nav-link" data-scroll-target="#discussion-reinforcement"><span class="header-section-number">6</span> Discussion: Reinforcement</a>
  <ul class="collapse">
  <li><a href="#bemmels-et-al-2021" id="toc-bemmels-et-al-2021" class="nav-link" data-scroll-target="#bemmels-et-al-2021"><span class="header-section-number">6.1</span> Bemmels et al 2021</a></li>
  <li><a href="#levitan-hao-2025" id="toc-levitan-hao-2025" class="nav-link" data-scroll-target="#levitan-hao-2025"><span class="header-section-number">6.2</span> Levitan &amp; Hao 2025</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Reading notes</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="mallet-2013" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Mallet 2013</h1>
<ul>
<li></li>
</ul>
</section>
<section id="foote-2018" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Foote 2018</h1>
<ul>
<li>Sympatric speciation does not involve physical/geographic barriers, so offers insights into role of selection (natural and sexual) in driving speciation.</li>
<li>This speciation (divergence with gene flow) is therefore rare, extreme, restrictive, and controversial.</li>
<li>Genomic methods have provided evidence of two kinds: either multiple colonisations and secondary contact, or actual sympatric divergence.</li>
<li>Author overall argues for less focus on primary vs secondary contact as driving speciation, and more on mechanistic aspects of genomic architecture.</li>
<li>Previous methods used phylogenetics to infer sympatric speciation from monophyly, but monophyly is not a good indicator of sympatric speciation (consistent with, but not exclusive to). (One flaw of Coyne &amp; Orr’s criteria.)</li>
<li>Backwards-in-time approaches are limited to identifying gene flow events arising through coalescence. They cannot distinguish between periods of spatial isolation and of spatial overlap with on gene flow.</li>
<li>In cases of actual sympatric speciation, we expect barrier loci to be clustered tightly (close together) in the genome, and at areas of low recombination, and to have a “large effect” especially affecting both reproductive isolation and ecological adaptation. That is, <em>linkage</em> or <em>pleiotropy</em>. Otherwise, ongoing gene flow (recombination, genome-wide homogenisation) would make it difficult for divergence.</li>
<li>Modern genomic methods are still biased towards disproving (and not supporting) primary sympatric speciation. Palaeogenomics is a useful tool to tackle this.</li>
<li>Author argues that focus should be placed on sympatric speciation, and away from primary vs secondary.</li>
</ul>
</section>
<section id="discussion-geography-of-speciation" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Discussion: Geography of speciation</h1>
<section id="coathup-et-al-2024" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="coathup-et-al-2024"><span class="header-section-number">3.1</span> Coathup et al 2024</h2>
<ul>
<li>Theory says sympatric speciation facilitated by magic traits which might be encoded by pleiotropic genes, but little evidence exists of sympatric speciation and empirical evidence of magic traits is rare. Authors investigated role of pleitropy in <em>Howea</em> speciation. They found multiple genes showing adaptive divergence between the sister species (sign of pleiotropy, in soil stress tolerance and flowering time).</li>
<li>Alternative to magic traits/pleiotropy driving sympatric ecological speciation is linkage disequilibrium. (i.e., one allele with multiple effects vs two close alleles not separated in recombination.) In either case, “major effect loci”.</li>
<li>Dunning et al 2016 and Osborne et al 2019 assessed gene expression and genetic divergence. They identified candidate pleiotropic speciation genes. The current study aimed to test whether these genes actually are pleiotropic loci affecting soil stress tolerance and flowering time. If so, this would be positive evidence for magic traits.</li>
<li>Method: T-DNA knockout <em>A. thaliana</em> used for high-throughput phenotyping (knockouts in orthologues of candidate speciation genes, 11 knockout mutants).
<ul>
<li>Grown under various stress emulating Lord Howe Island. Then morphology, stress responses and flowering time measured.</li>
</ul></li>
<li>Five genes potential speciation genes: knockouts in each of these showed displaced flowering times and altered soil stress response (cadmium, drought, salt). Largest effect sizes in SAL1 (latest flowering) and SIZ1 (earliest flowering)
<ul>
<li>Knockout of SIZ1 caused reduced drought tolerance. <em>H. belmoreana</em> has lower expression of the two SIZ1 homologues and also cannot colonise calcareous soils. On the other hand, knockout of SAL1 caused increased drought tolerance (consistent with <em>H. forsteriana</em>), but reduced salt tolerance which doesn’t match with <em>Howea</em> scenario—so less evidence for this locus driving sympatric speciation cf SIZ1.</li>
</ul></li>
<li>Robust example of genetic architecture of sympatric speciation. Evidence of pleiotropy in multiple loci, so could be a combination of pleiotropy and linkage disequilibrium.</li>
</ul>
<p>How did knockout (removal) give these results? Counterintuitive?</p>
</section>
<section id="dean-et-al-2019" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="dean-et-al-2019"><span class="header-section-number">3.2</span> Dean et al 2019</h2>
<ul>
<li>An assumption when considering sympatric species pairs is that they originated from a recent, genetically homogenous ancestor, but most evidence invokes introgression/admixture upon secondary contact as responsible for speciation. The authors compare two ecotypes of three-spined stickleback (no, multiple species-pairs?) using morphological data, targeted genotyping and genome-wide SNP data.
<ul>
<li>The two ecotypes show extensive hybridisation in sympatry/parapary.</li>
<li>There are rare cases of complete reproductive isolation in sticklebacks, but these cases involve ancient allopatric origins.</li>
</ul></li>
<li>Importance of genetic admixture upon secondary contact widely known in plants since 1970s, but has also started gaining popularity in animals.
<ul>
<li>Classic examples like Darwin’s finches, Lake Victoria cichlids and <em>Anopheles</em> species complex all either involved genetic admixture events or are cryptic cases of homoploid hybrid speciation.</li>
</ul></li>
<li>Overwhelming evidence for ancient allopatry, and against “true” sympatric ecological speciation. (And authors found the same.)</li>
<li>They find that the two ecotypes are strongly reproductively isolated, but also a complex colonisation history suggesting admixture upon secondary contact in addition to disruptive ecological selection (i.e., older allopatric origin).</li>
<li>Three hypotheses (none consider sympatric speciation option? wait, the first classic one probably is)</li>
<li>Compelling morphological and genetic evidence of unusually strong reproductive isolation between lagoon resident and anadromous stickleback, in spite of (low) gene flow. Classic purely-ecological speciation model (relative sea level change) unlikely; at least two colonisation and therefore admixture upon secondary contact (alongside strong selection for maintenance of differentiation, i.e.&nbsp;partial ecological basis)</li>
<li>Reproductive isolation in this island species-pair example is considerably stronger than most other stickleback examples, so this could potentially be a very useful system to work further on these questions.</li>
<li>Also evidence against “linear, bifurcating process” model.</li>
</ul>
<p>Species pairs? Or just ecotypes? Confusing.</p>
</section>
</section>
<section id="rundle-nosil-2005" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Rundle &amp; Nosil 2005</h1>
<ul>
<li>Renewed interest in microevolutionary processes driving macroevolutionary speciation was linked to reclassficiation of speciation models from geographic scheme (sympatry vs allopatry) to focus on reproductive isolation.</li>
<li>Barriers as a result of ecologically driven divergent selection</li>
<li>Ecological speciation different from other models that focus on chance events (polyploidisation, hybridisation, genetic drift, founder events, population bottlenecks), or on non-divergent and/or non-ecological selection (runaway selection, sexual conflict)</li>
</ul>
<section id="ecological-causes-of-divergent-selection" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="ecological-causes-of-divergent-selection"><span class="header-section-number">4.1</span> Ecological causes of divergent selection</h2>
<p>Environmental differences, sexual selection, ecological interactions</p>
</section>
</section>
<section id="discussion-pre-zygotic-isolation" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Discussion: Pre-zygotic isolation</h1>
<section id="zhang-et-al-2024" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="zhang-et-al-2024"><span class="header-section-number">5.1</span> Zhang et al 2024</h2>
<ul>
<li>Role of ecology and geography have not been explored much at later stages of speciation, because traditionally focus on role of ecology in <em>initiating</em> speciation and on role of geography (reinforcement) on species pairs lacking divergent characteristics.</li>
<li>How divergent ecology interacts with geography to affect RI in later stages of speciation remains unclear</li>
<li>Reinforcement: increased favour towards prezygotic isolation in sympatry, peripatry or parapatry, due to fitness cost to hybridisation</li>
<li>Increased habitat isolation can also arise from direct selection against immigrants (high fidelity to native environment favoured)</li>
<li>Three gall wasps in genus <em>Belonocnema</em>, specific to <em>Quercus</em> hosts.</li>
<li>Existing knowledge: two sympatric wasp species are in later stages of speciation</li>
<li>Test: in allopatry, does host plant divergence yield higher RI? In host plant divergence, does sympatry yield higher RI than allopatry (due to selection for reduced migration and reduced hybridisation)?</li>
<li>Measured “degree of sexual isolation in the absence of host plant”</li>
<li>Natural temporal difference in the wasp species, so they stored some galls at 4 dec C for one week which delayed the emergence. Adults were used in experiments 1-2 days after emergence.</li>
<li>Measured host plant selection using a controlled two-choice experiment, using females instead of males (female host choice determines environment for next generation, and males display lower host fidelity).
<ul>
<li>Female aspirated into the container and observed for 30 min at 2-min intervals (15 obs), with the assumption that time spent on host plant is correlated with oviposition decision.</li>
<li>Calculated proportion of time spent on natal plant (<span class="math inline">\([0, 1]\)</span>), compared using GLMM.</li>
<li>Also calculated pairwise estimates of host plant preference difference: absolute difference between preference values calculated above (also <span class="math inline">\([0, 1]\)</span>).</li>
</ul></li>
<li>Measured sexual isolation by observing courtship and mating interactions between males and females, using controlled no-choice preference trials.
<ul>
<li>Male and female (same or different species) aspirated into container, observed for same time, and recorded three interactions: male wing buzzing/fanning, male mounting female, copulation.</li>
<li>Calculated probability of mating between conspecific and heterospecific pairs, and compared using GLMM.</li>
<li>Comparison between species pair type and individual pair cannot be done using GLMM, so they also calculated strength of sexual isolation between species pairs using a formula.</li>
</ul></li>
<li>Evidence that host use (ecological) interacts with geography in driving divergent selection in <em>Belonocnema</em> wasps.
<ul>
<li>Sympatric wasp species with different host plant showed highest habitat isolation, while allopatric with same host plant showed lowest.</li>
<li>Proximity to two host plant options increased host preference (Petri dish vs cups)</li>
<li>Sympatric also showed highest sexual isolation, while allopatric with same host plant showed lowest.</li>
</ul></li>
<li>High isolation in allopatric species with same host plant suggest that selection is not directly driven by habitat use. Similar selection environment could have taken different phenotypic pathways (mutation-order speciation).</li>
<li>Three possible mechanisms for high isolation in sympatric species:
<ul>
<li>Templeton effect/differential fusion: in sympatry, strong isolation is required for isolated populations to persist, whereas weakly isolated populations fuse.</li>
<li>Reinforcement: natural selection against hybrids with lower fitness</li>
<li>Reproductive interference: presence of sympatric species during mating (or cost of migration) could elevate prezygotic isolation</li>
</ul></li>
<li>Also found asymmetry between habitat isolation and sexual isolation between allopatric species sharing the same host plant: lineage that emigrates more shows stronger habitat isolation (lower fitness of immigrants in alternative habitat) while lineage that receives more immigrants shows stronger sexual isolation (lower fitness of hybrids).</li>
<li>The three closely related wasp species are a good model for speciation research, due to their unique ecology and geography. Their framework is also useful for exploring completion of speciation, a knowledge gap.</li>
</ul>
</section>
<section id="schield-et-al-2024" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="schield-et-al-2024"><span class="header-section-number">5.2</span> Schield et al 2024</h2>
<ul>
<li>How does the genetic basis of sexual selection result in barrier to gene flow in secondary contact?</li>
<li>Most studies explore links between either sexual traits and reproductive isolation, or genomic divergence and genetic association with sexual traits. Not the full flow from genomic divergence to reproductive isolation via barrier to gene flow.</li>
<li>For species like Barn Swallow which are early in speciation process, sexual selection may be driving RI at small number of barrier loci involved in sexual traits.
<ul>
<li>Genetic coupling of barrier loci (linkage disequilibrium) facilitates speciation, via both direct and indirect selection. With increase in coupled barrier loci, barrier to gene flow is stengthened.</li>
<li>Coupling is likely to arise in allopatry, and can promote RI upon further secondary contact.</li>
</ul></li>
<li>Barn Swallow populations all diverged recently from the common ancestor about 11 kya, and there is strong evidence for divergent sexual selection among these populations.
<ul>
<li>Not only does the species have multiple populations (subspecies), but there are also multiple hybrid zones, with varying divergence in sexual traits. These hybrid zones allow tests of whether sexual selection results in RI upon secondary contact.</li>
<li>Further, two of these hybrid zones also have divergences in nonbreeding migratory route, which is a potential coupled trait (low survival of hybrids taking nonadaptive migratory routes; i.e., hybrid fitness).</li>
<li>Explicit tests of effects of trait loci on gene flow are necessary because degree of concordance between allopatric evolutionary processes and hybrid fitness is not clear a priori.</li>
</ul></li>
<li>Do sexual trait loci experience divergent sexual selection in allopatry and promote RI upon secondary contact? Is there LD among barrier loci (sexual traits, migratory route) in secondary contact, mainting RI via effects on hybrid fitness?</li>
<li>Found that 6 ssp. cluster in 3 groups. Individuals from hybrid zones show intermediate/similar to parent plumage trait values (breast feather colour, tail streamer length) and admixture proportions.</li>
<li>Prevalence of back-crossed, late generation hybrids, and demographic models, suggest that hybrid zones formed upon secondary contact following allopatry.</li>
<li><em>rustica</em>-<em>gutturalis</em> hybrid zone estimated to have formed 2000 generations ago, while the <em>tytleri</em> ones were more recent (&lt; 1000 generations).</li>
<li>Trait variation is controlled by few loci and not several genes: both traits have high proportion of phenotypic variation explained (PVE) by SNP genotypes, and high proportion of genetic effects (PGE) by small number of SNPs.</li>
<li>Ventral colour SNPs on chromosome 1A and Z (genes affecting melanogenesis and feathering phenotype), whereas tail streamer length SNPs on chromosome 2 (enzymes with regulatory functions potentially affecting transcription, intracellular transport, growth factor stability).
<ul>
<li>Concentration of ventral colour associations on Z chromosome (sex chromosome) supports that sex linkage is favourable for traits involved in mate choice and sexual isolation.</li>
</ul></li>
<li>Across allopatric populations, sexual trait loci show more genetic differentiation (FST) than do genome-wide comparisons.
<ul>
<li>This could just be due to “background selection” in regions with reduced recombination. However, regions with extremely low recombination rates like centromeres also show higher FST in trait loci—high differentiation in trait loci is due to divergent positive selection rather than background selection alone. Elevated allelic differentitation and extended haplotype homozygosity, also suggest this.</li>
<li>Greater fitness effects of ventral colour than tail streamer length.</li>
</ul></li>
<li>Geographic and genomic cline analysis, comparing trait loci–background loci differences across space along the three hybrid zone transects. -Found steeper transition between parental ancestries in trait loci cf background loci.
<ul>
<li>Steepness itself could arise due to low population density in hybrid zones, but this would show similar clines in trait loci as in other highly differentiated loci (not the case)</li>
<li>Stronger selection and barrier to gene flow between <em>rustica</em> and other ssp.</li>
<li>Cline patterns are weakest in <em>tytleri</em>-<em>gutturalis</em> hybrid zone, possibly linked to absence of migratory divide between the two populations (though remains to be tested).</li>
</ul></li>
<li>LD arises from hybridisation, but extent of erosion of LD by recombination over generations can be impacted by selection-driven genetic coupling.
<ul>
<li>They compared interchromosomal LD between trait loci and background loci; if the LD was formed by admixture but not maintained by selection, would find similar levels between the two. But they found much higher LD levels in trait loci.</li>
<li>Increase in LD shows consistent differences between hybrid zones, with <em>rustic</em> hybrids showing higher.</li>
<li>Also no differences in LD between parental populations, therefore admixture important followed by selection.</li>
<li>Starting levels of admixture LD higher than current levels, suggesting that selection worked by slowing erosion of LD, rather enhancing LD beyond starting levels.</li>
</ul></li>
<li>Primary/major role of divergent sexual selection in genomic divergence and RI.</li>
</ul>
</section>
</section>
<section id="discussion-reinforcement" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Discussion: Reinforcement</h1>
<section id="bemmels-et-al-2021" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="bemmels-et-al-2021"><span class="header-section-number">6.1</span> Bemmels et al 2021</h2>
<ul>
<li>Dynamics of allopatry initiating speciation, widely accepted. But how sympatry (sec.&nbsp;contact) can complete speciation is contested.</li>
<li>RI before secondary contact: either fully complete, or costs to interactions between partially RI species (maladaptive hybridisation) may drive further divergence (reinforcement).</li>
<li>Problem with complete RI prior to sympatry: takes very long time to evolve, and widepsread occurrence of hybrid zones suggests frequent contact with incomplete RI.</li>
<li>Alternative is that incipient species interactions change (indirect)—any process that results in divergent character displacement, e.g., divergent agonistic selection to avoid aggressive interactions (though this could result in the opposite, convergence, if interspecific aggression helps defend against true competitors).</li>
<li>Another option is direct selection for increased RI (direct): reinforcement, i.e., selection to avoid maladaptive hybridisation. Very difficult to demonstrate its existence, so few examples. One of the hallmark indicators of reinforcement is reproductive character displacement (increased divergence in reproductive traits in sympatry than in allopatry), but this can be generated by other processes too.</li>
<li>Difficult to determine whether sympatry plays a role in completion of speciation (geographic context difficult to estimate).</li>
<li>Coalescent demographic modelling, whole-genome datasets. Sympatric E NA <em>Empidonax</em> species populations (broad sympatry) show increased RI post secondary contact—unrelated to reproductive character displacement—while western contact zone (narrow sympatry) has lower RI (higher introgression).</li>
<li>Methods
<ul>
<li>Whole genome data –&gt; patterns of admixture (is RI higher in broad sympatry than in narrow sympatry?), and patterns of ecological/reproductive character displacement in sympatry. Coalescent modelling –&gt; did geographic contact precede and drive increase in RI?</li>
<li>Blood samples –&gt; whole-genome sequencing and genotyping-by-sequencing –&gt; alignment with Willow reference genome, generation of de no Alder genome (linked-read sequencing).</li>
<li>Genomic differentiation: weighted Hudson’s F_ST_, proportion of genetic ancestry.</li>
<li>Geographic contact: GBIF, WorldClim, Maxent SDMs for present climate and LGM. (Potential for range overlap.)</li>
<li>Character displacement: morphological measurements of caught birds. Temporal displacement from spring arrival time from eBird.</li>
</ul></li>
<li>Genomes are well differentiated, multiple F_ST_ peaks on most chromosomes, greater in east than west. Similarly, admixture lower in east than west. RI in east not absolute but nearly complete and significantly higher than in west.</li>
<li>Longer contact in east than in west, south of glaciation. Willow occurred in both but Alder (currently more N breeder) only in east. (However, range overlap in LGM not really in “east”, more central [Fig 3].)</li>
<li>No temporal character displacement. No displacement in bill/body morphology or crown feather reflectance. Only difference in crown colour—subtle, but strongest in UV, but birds unlikely to perceive this difference.</li>
<li>Support for continuous gene flow even in present day in west, but ceased long after secondary contact in east. However, model support relatively weaker in west (additional complexity).</li>
<li>Elevated RI could have been due to ecological divergence to avoid resource competition, agonistic selection, or reinforcement—but causal traits elusive.</li>
<li>Multistage model of speciation: dynamics of initiation and completion of RI have fundamentally different dynamics and drivers.</li>
<li>Potential alternative reproductive traits that could be under divergent selection: male song, and female preference for male song. Is divergence of male song stronger in sympatry? Is female preference for conspecific song higher in sympatry (reinforcing selection) as in <em>Ficedule</em>? Or even gametic level (sperm-related). Also need to explore costs to hybridisation like reduced fitness.</li>
<li>Their approach is one effective way to study what otherwise remains very difficult, even in non-model systems and species that no longer hybridise.</li>
<li>Broad sympatry vs narrow sympatry —&gt; sympatry vs allopatry (not sure if this logical jump is appropriate)</li>
</ul>
</section>
<section id="levitan-hao-2025" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="levitan-hao-2025"><span class="header-section-number">6.2</span> Levitan &amp; Hao 2025</h2>
<p>About postzygotic (gametic) isolation</p>
<ul>
<li>How do initially rare reproductive genotypes proliferate over time if they by definition result in reduced mate compatibility?</li>
<li>External fertilisation –&gt; ease of study, wide variation in intraspecific gametic compatibility.</li>
<li>Possibility: sexual conflict over fertilisation rate selects for eggs with reduced compatibility, producing population with more than one matched compatibility group. But although this generates variation, not sufficient to explain very low compatibility preventing/limiting hybridisation.</li>
<li>Two urchin species can produce F1 embryos; <em>Sd</em> eggs more susceptible than <em>Mf</em>; former can metamorphose into juveniles but magnitudes lower than conspecifics (no introgression between the two species).</li>
<li>Tested for conspecific sperm precedence (CSP) using no-choice and competitive choice assays. Is variation in hybrid fertilisation related to male and female gamete proteins? Are derived protein variants less likely to result in hybrid fertilisation than ancestral? Do allele frequencies of sperm and egg proteins in sympatry and allopatry support presence of RCD?</li>
<li>No-choice fertilisation success between conspecific and heterospecific males and females reciprocally. Choice experiments using <em>Sd</em> eggs, because it has higher and more variable levels of hybridisation than <em>Mf</em>.</li>
<li>Experiment 1: no-choice + 5 choice (varying heterospecific sperm ratios) crosses. Experiment 2: no-choice + 0.9:0.1 (<em>Mf</em>:<em>Sd</em>) choice crosses (ability of sperm to fertilise eggs at greatest numerical superiority; more replicates).</li>
<li>In choice crosses, embryos developed for 3 days then genetic analysis for paternity. Survivorship compared among crosses to test for bias between early postzygotic survivorship (prior to genotyping larvae) and gametic incompatibility.</li>
<li><em>Sd</em> more susceptible to hybrid fertilisation than <em>Mf</em>, and required lower concentration of heterspecific sperm. But presence of conspecific sperm reduced hybridisation—47% no hybridisation even at most extreme ratios favouring hybridisation. Concentration no longer important when both sperm present; instead, relative proportion important.</li>
<li>Above were overall/average trends, but individual results varied from absolute to nonexistent CSP. This variation was explained by the sperm genotype: presence of shared (ancestral) GG allele increased hybridisation. Higher hybridisation in presence of shared protein was seen also in egg protein, but this was not statistically significant.</li>
<li>Shared allele more frequent in allopatry than sympatry, but only in eggs and not in sperm bindin (greater cost of egg wastage than sperm wastage).</li>
<li>Interspecific variation in sperm recognition proteins not enough to fully prevent hybrid fertilisation if given enough time and without competing sperm with higher affinity (conspecifics).</li>
<li>Likely that reinforcement selection and RCD manifested more in <em>Mf</em>.</li>
<li>Sexual conflict, or generally sexual selection, generates intraspecific variation which then gets acted upon by reinforcement selection.</li>
<li>Did not like the way this paper was written:
<ul>
<li>Background on the proteins: important info but could have had lay summaries too.</li>
<li>Overll just too technical!</li>
<li>Results too long, longer than discussion, and contains things that should be in methods (like models and tests)!</li>
<li>Paper should overall have been more inferential, rather than technical reporting.</li>
</ul></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>